<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="ChIPanalyser">
<title>The ChIPanalyser User's Guide • ChIPanalyser</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.1.3/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.1.3/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.rawgit.com/afeld/bootstrap-toc/v1.0.1/dist/bootstrap-toc.min.js"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="The ChIPanalyser User's Guide">
<meta property="og:description" content="ChIPanalyser">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">ChIPanalyser</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">2.0.1</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="active nav-item">
  <a class="nav-link" href="../articles/ChIPanalyser.html">Get started</a>
</li>
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/GA_ChIPanalyser.html">The ChIPanalyser User's Guide</a>
  </div>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav"></ul>
</div>

    
  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>The ChIPanalyser User's Guide</h1>
                        <h4 data-toc-skip class="author">Patrick
Martin</h4>
            
      
      
      <div class="d-none name"><code>ChIPanalyser.Rmd</code></div>
    </div>

    
    
<div class="section level2">
<h2 id="introduction---what-is-this-package">Introduction - What is this package?<a class="anchor" aria-label="anchor" href="#introduction---what-is-this-package"></a>
</h2>
<p>ChIPanalyser provides a quick an easy method to predict and explain
TF binding. The package uses a statistical thermodynamic framework to
model the bidning of proteins to DNA. ChIPanalyser makes the assumption
that the probability that any given sites along the genome is bound by a
TF can be explained by four main factors: DNA accessibility, Binding
energy, a scaling factor modulating binding energy (<span class="math inline">\(\lambda\)</span>) and number of TF bound (<span class="math inline">\(N\)</span>) to DNA. Both <span class="math inline">\(N\)</span> and <span class="math inline">\(\lambda\)</span> are inferred internally by
maximisng (or minimising) a goodness of fit metric between predictions
and actual ChIP-data. The package will produce a ChIP like profile at a
base pair level. As opposed to machine learning type frameworks, if
these paramters are known by other means (experimentally), ChIPanalyser
does require any training to produce ChIP like profiles. Estimated
values can directly be plugged into the model. Furthermore, ChIPanalyser
helps gain an understanding of the mechanims behind TF binding as
described by (Zabet et al 2015 &amp; Martin and Zabet 2019).</p>
</div>
<div class="section level2">
<h2 id="methods---the-model">Methods - The Model<a class="anchor" aria-label="anchor" href="#methods---the-model"></a>
</h2>
<p>As described above, ChIPAnalyser is based on an approximation of
statistical thermodynamics. The core formula describing TF binding is
given by : <span class="math display">\[P(N,a,\lambda,\omega)_{j} =
\frac{N \cdot a_{j} \cdot e^{(\frac{1}{\lambda}
\cdot \omega_{j})}}{N \cdot a_{j} \cdot e^{(\frac{1}{\lambda}
\cdot \omega_{j})}+ L \cdot n \cdot [a_{i} \cdot e^{(\frac{1}
\lambda \cdot \omega_j)}]_ {i}} \]</span> with</p>
<ul>
<li>N , the number of TF molecules bound to DNA</li>
<li>a , DNA accessibility</li>
<li>
<span class="math inline">\(\lambda\)</span> , a parameter scaling
the specificity of a given TF</li>
<li>
<span class="math inline">\(\omega\)</span> , a Position Weight
Matrix.</li>
</ul>
</div>
<div class="section level2">
<h2 id="work-flow">Work Flow<a class="anchor" aria-label="anchor" href="#work-flow"></a>
</h2>
<p>The next section will be split between the following subsections</p>
<ul>
<li>
<strong>Loading Data</strong> - Description of internal Data. We
will use this data for our work flow example.</li>
<li>
<strong>Quick Start</strong> - We will give quick start example.
Only core functionalities and work flow will be described in this
section.</li>
<li>
<strong>Advanced Work</strong> - We will describe a more indepth
work flow.</li>
<li>
<strong>Parameter Description</strong> - We will give an in depth
description of each parameter used in ChIPAnalyser</li>
</ul>
<div class="section level3">
<h3 id="loading-data">Loading Data<a class="anchor" aria-label="anchor" href="#loading-data"></a>
</h3>
<p>Before going through the inner workings of the package and the work
flow, this section will quickly demonstrate how to load example datasets
stored in the package. This data represents a minimal workable examples
for the different functions. All data is derived from real biological
data in <em>Drosophila melanogaster</em> (The <em>Drosophila
melanogaster</em> genome can be found as a <code>BSgenome</code> ).</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">ChIPanalyser</span><span class="op">)</span></span>
<span></span>
<span><span class="co">#Load data</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">ChIPanalyserData</span><span class="op">)</span></span>
<span></span>
<span></span>
<span><span class="co"># Loading DNASequenceSet from BSgenome object</span></span>
<span><span class="co"># We recommend using the latest version of the genome </span></span>
<span><span class="co"># Please ensure that all your data is aligned to the same version of the genome</span></span>
<span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">BSgenome.Dmelanogaster.UCSC.dm6</span><span class="op">)</span></span>
<span></span>
<span><span class="va">DNASequenceSet</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/Biostrings/man/getSeq.html" class="external-link">getSeq</a></span><span class="op">(</span><span class="va">BSgenome.Dmelanogaster.UCSC.dm6</span><span class="op">)</span></span>
<span></span>
<span></span>
<span></span>
<span><span class="co">#Loading Position Frequency Matrix</span></span>
<span></span>
<span><span class="va">PFM</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>,package<span class="op">=</span><span class="st">"ChIPanalyser"</span><span class="op">)</span>,<span class="st">"BEAF-32.pfm"</span><span class="op">)</span></span>
<span></span>
<span><span class="co">#Checking if correctly loaded</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/ls.html" class="external-link">ls</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] "Access"         "chip"           "cs"             "DNASequenceSet"</span></span>
<span><span class="co">## [5] "geneRef"        "PFM"            "top"</span></span></code></pre>
<p>The global environment should now contain a few new variables:
DNASequenceSet, PFM, Access, geneRef, top, chip.</p>
<ul>
<li><p><code>DNASequenceSet</code> is DNAStringSet extracted from the
<em>Drosophila melanogaster</em> genome (BSgenome). It is advised to use
a full genome sequence for this object.</p></li>
<li><p><code>PFM</code> is a path to file. In this case, it is a
Position Frequency Matrix derived from the Bicoid Transcription factor
in <em>Drosophila melanogaster</em>in <em>RAW</em> format. we provide
loading support for JASPAR, Transfac and RAW. If you wish to use any
other format, we suggest to use the MotifDb package (or load PFM as
matrix into R) and parse the matrix to ChIPanalyser. In this scenario,
<code>PFMFormat</code> argument should be set to <em>matrix</em> (see
below for more information).</p></li>
<li><p><code>Access</code> is a <code>GRanges</code> object containing
accessible DNA for the sequence above.</p></li>
<li><p><code>geneRef</code> is a <code>GRanges</code> containing gene
reference annotations.</p></li>
<li><p><code>top</code> is a <code>GRanges</code> object with 4 genomic
regions containing BEAF-32 binding.</p></li>
<li><p><code>chip</code> is a <code>GRanges</code> with ChIP score for
the BEAF-32 Transcription Factor.</p></li>
</ul>
<p><strong>IMPORTANT:</strong> Data sets provided in the package have
been currated to meet the size requirements for Bioconductor packages.
Please read the instruction below carefully as we will describe how to
incorportate your own data into the pipe line.</p>
<p><strong>NOTE</strong> You can download gene reference from (Genome
UCSC website)[<a href="https://genome.ucsc.edu/" class="external-link uri">https://genome.ucsc.edu/</a>]</p>
</div>
<div class="section level3">
<h3 id="quick-start">Quick Start<a class="anchor" aria-label="anchor" href="#quick-start"></a>
</h3>
<div class="section level4">
<h4 id="step-1---extracting-normalised-chip-scores-from-chip-seq-datasets">Step 1 - Extracting Normalised ChIP scores from ChIP-seq
datasets<a class="anchor" aria-label="anchor" href="#step-1---extracting-normalised-chip-scores-from-chip-seq-datasets"></a>
</h4>
<p>ChIPAnalyser requires ChIP data in order to infer the optimal set of
values that will be assigned to bound Molecules and lambda. The package
will maximise (or minimise) a goodness of fit metric between the
predictions and ChIP data.</p>
<p>If you have inferred or approximated the values to be assigned to
<span class="math inline">\(N\)</span> and <span class="math inline">\(\lambda\)</span> by other means, skip this step
and go directly to <strong>Advanced Work</strong>.</p>
<p><code>chip</code> can be a connection to your ChIP data, a GRanges of
your ChIP or data frame (<em>bed</em> format style) <code>loci</code> is
a GRanges object containing loci of interest. Default set as NULL (see
<strong>Advanced Work</strong>)</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">chip</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/processingChIP.html">processingChIP</a></span><span class="op">(</span>profile <span class="op">=</span> <span class="va">chip</span>,</span>
<span>                      loci <span class="op">=</span> <span class="va">top</span>,</span>
<span>                      cores <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span>
<span><span class="va">chip</span></span></code></pre></div>
<pre><code><span><span class="co">## _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ </span></span>
<span><span class="co">##                         ChIP score from 4 regions </span></span>
<span><span class="co">## _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ </span></span>
<span><span class="co">## [ 1 ]  chr3L:20300001..20320000 [ 2 ]  chr2L:6480001..6500000 [ 3 ]  chr3R:16140001..16160000 [ 4 ]  chr2L:19500001..19520000 </span></span>
<span><span class="co">## _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ </span></span>
<span><span class="co">##  </span></span>
<span><span class="co">## _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ </span></span>
<span><span class="co">##  </span></span>
<span><span class="co">##                         Top  4  regions </span></span>
<span><span class="co">## _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ </span></span>
<span><span class="co">## GRanges object with 4 ranges and 0 metadata columns:</span></span>
<span><span class="co">##                            seqnames            ranges strand</span></span>
<span><span class="co">##                               &lt;Rle&gt;         &lt;IRanges&gt;  &lt;Rle&gt;</span></span>
<span><span class="co">##   chr3L:20300001..20320000    chr3L 20300001-20320000      *</span></span>
<span><span class="co">##     chr2L:6480001..6500000    chr2L   6480001-6500000      *</span></span>
<span><span class="co">##   chr3R:16140001..16160000    chr3R 16140001-16160000      *</span></span>
<span><span class="co">##   chr2L:19500001..19520000    chr2L 19500001-19520000      *</span></span>
<span><span class="co">##   -------</span></span>
<span><span class="co">##   seqinfo: 3 sequences from an unspecified genome; no seqlengths</span></span>
<span><span class="co">## _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ </span></span>
<span><span class="co">##  </span></span>
<span><span class="co">## _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ </span></span>
<span><span class="co">##  </span></span>
<span><span class="co">##                         Associated Options </span></span>
<span><span class="co">## _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ </span></span>
<span><span class="co">##  backgroundSignal:                           0.0354659475382993 </span></span>
<span><span class="co">##  maxSignal:                                                   1 </span></span>
<span><span class="co">##  chipMean:                                                  200 </span></span>
<span><span class="co">##  chipSd:                                                    200 </span></span>
<span><span class="co">##  chipSmooth:                                                250 </span></span>
<span><span class="co">##  lociWidth:                                               20000 </span></span>
<span><span class="co">## _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _</span></span></code></pre>
<p>The output of <code>processingChIP</code> returns a
<code>ChIPScore</code> object containing extracted and normalised ChIP
scores, your loci of interest and internal parameters associated to the
ChIP extractions process.</p>
</div>
<div class="section level4">
<h4 id="step-2---computing-a-pwms">Step 2 - Computing a PWMs<a class="anchor" aria-label="anchor" href="#step-2---computing-a-pwms"></a>
</h4>
<p>The model relies on a Postion Weight Matrix.
<code>genomicProfiles</code> serve as a way of storing important
paramters. More importantly, this object stores intermediate results as
you make your way through the analysis pipeline.</p>
<p>From a Position Frequency Matrix:</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># PFMs are automatically converted to PWM when build genomicProfiles</span></span>
<span><span class="va">GP</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/genomicProfiles.html">genomicProfiles</a></span><span class="op">(</span>PFM <span class="op">=</span> <span class="va">PFM</span>, PFMFormat <span class="op">=</span> <span class="st">"JASPAR"</span><span class="op">)</span></span>
<span><span class="va">GP</span></span></code></pre></div>
<pre><code><span><span class="co">## _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ </span></span>
<span><span class="co">##  </span></span>
<span><span class="co">##                         genomicProfiles object: Position Weight Matrix </span></span>
<span><span class="co">## _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ </span></span>
<span><span class="co">##  </span></span>
<span><span class="co">## Position Weight Matrix (PWM): </span></span>
<span><span class="co">##  </span></span>
<span><span class="co">##         [,1]        [,2]       [,3]       [,4]         [,5]        [,6]</span></span>
<span><span class="co">## A -0.2891885 -0.03809741  0.2316372  0.6301128  0.687738702  0.79025750</span></span>
<span><span class="co">## C -0.1331731  0.36623770 -0.1622533  0.3097032 -1.062881002 -1.12273432</span></span>
<span><span class="co">## G  0.1277129 -0.16225333  0.2729606 -0.2758771 -0.395107494  0.01666913</span></span>
<span><span class="co">## T  0.2147795 -0.29456303 -0.5529272 -8.2905435 -0.008313437 -0.79001402</span></span>
<span><span class="co">##         [,7]       [,8]      [,9]     [,10]     [,11]     [,12]      [,13]</span></span>
<span><span class="co">## A  0.4399853 -0.2905294  1.366596 -8.290544 -8.290544 -2.557202  1.2881138</span></span>
<span><span class="co">## C  0.6139508 -0.4502305 -8.290544 -8.290544  1.386106 -8.290544 -8.2905435</span></span>
<span><span class="co">## G -0.5119134 -8.2905435 -2.557202 -8.290544 -8.290544  1.366596 -0.9846835</span></span>
<span><span class="co">## T -8.2905435  0.9610348 -8.290544  1.386106 -8.290544 -8.290544 -8.2905435</span></span>
<span><span class="co">##       [,14]       [,15]</span></span>
<span><span class="co">## A -0.844542  0.77465549</span></span>
<span><span class="co">## C -1.766981 -8.29054350</span></span>
<span><span class="co">## G -8.290544 -0.23443384</span></span>
<span><span class="co">## T  1.223525  0.03814908</span></span>
<span><span class="co">## _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ </span></span>
<span><span class="co">##  </span></span>
<span><span class="co">## Base pairs frequency for PWM weighting : </span></span>
<span><span class="co">##  </span></span>
<span><span class="co">##    A    C    G    T </span></span>
<span><span class="co">## 0.25 0.25 0.25 0.25 </span></span>
<span><span class="co">## _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ </span></span>
<span><span class="co">##  </span></span>
<span><span class="co">## PWM built from (PFM - Format  JASPAR ) : </span></span>
<span><span class="co">##  </span></span>
<span><span class="co">##   [,1] [,2] [,3] [,4] [,5] [,6] [,7] [,8] [,9] [,10] [,11] [,12] [,13] [,14]</span></span>
<span><span class="co">## A  746  959 1256 1871 1982 2196 1547  745 3908     0     0    77  3613   428</span></span>
<span><span class="co">## C  872 1437  847 1358  344  324 1841  635    0     0  3985     0     0   170</span></span>
<span><span class="co">## G 1132  847 1309  756  671 1013  597    0   77     0     0  3908   372     0</span></span>
<span><span class="co">## T 1235  742  573    0  988  452    0 2605    0  3985     0     0     0  3387</span></span>
<span><span class="co">##   [,15]</span></span>
<span><span class="co">## A  2162</span></span>
<span><span class="co">## C     0</span></span>
<span><span class="co">## G   788</span></span>
<span><span class="co">## T  1035</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##  _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _</span></span></code></pre>
<p>From a Position Weight Matrix:</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">GP</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/genomicProfiles.html">genomicProfiles</a></span><span class="op">(</span>PWM<span class="op">=</span><span class="va">PositionWeightMatrix</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="step-3---computing-optimal-parameters">Step 3 - Computing Optimal Parameters<a class="anchor" aria-label="anchor" href="#step-3---computing-optimal-parameters"></a>
</h4>
<p>As described above, ChIPanalyser infers the optimal combination of
bound molecules and lambda that will maximise (or minimise) a goodness
of fit metric. The following function computes the optimal set of
parameters and returns intermediate steps of the analysis pipeline. To
do so, we will be parsing the follwing: a DNA sequence, a Position
Weight Matrix (contained in a genomicProfiles), chromatin States
(Accessible DNA - This is optional), and extracted/normalised
experimental ChIP score (result of the <code>processingChIP</code>
function).</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## surpress dependency warnings</span></span>
<span><span class="va">optimal</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/warning.html" class="external-link">suppressWarnings</a></span><span class="op">(</span><span class="fu"><a href="../reference/computeOptimal.html">computeOptimal</a></span><span class="op">(</span>genomicProfiles <span class="op">=</span> <span class="va">GP</span>,</span>
<span>                        DNASequenceSet <span class="op">=</span> <span class="va">DNASequenceSet</span>,</span>
<span>                        ChIPScore <span class="op">=</span> <span class="va">chip</span>,</span>
<span>                        chromatinState <span class="op">=</span> <span class="va">Access</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Computing Genome Wide PWM Score</span></span></code></pre>
<pre><code><span><span class="co">## Computing PWM Score at Loci &amp; Extracting Sites Above Threshold</span></span></code></pre>
<pre><code><span><span class="co">## Computing Occupancy</span></span></code></pre>
<pre><code><span><span class="co">## Computing ChIP-seq-like Profile</span></span></code></pre>
<pre><code><span><span class="co">## Computing Accuracy of Profile</span></span></code></pre>
<p>NOTE Default Optimal parameters are provided internally as the
following:</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span> <span class="co">## Lambda Values</span></span>
<span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq</a></span><span class="op">(</span><span class="fl">0.25</span>,<span class="fl">5</span>,by<span class="op">=</span><span class="fl">0.25</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##  [1] 0.25 0.50 0.75 1.00 1.25 1.50 1.75 2.00 2.25 2.50 2.75 3.00 3.25 3.50 3.75</span></span>
<span><span class="co">## [16] 4.00 4.25 4.50 4.75 5.00</span></span></code></pre>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span> <span class="co">## Bound Molecule Values</span></span>
<span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fl">10</span>, <span class="fl">20</span>, <span class="fl">50</span>, <span class="fl">100</span>,</span>
<span>    <span class="fl">200</span>, <span class="fl">500</span>,<span class="fl">1000</span>,<span class="fl">2000</span>, <span class="fl">5000</span>,<span class="fl">10000</span>,<span class="fl">20000</span>,<span class="fl">50000</span>, <span class="fl">100000</span>,</span>
<span>    <span class="fl">200000</span>, <span class="fl">500000</span>, <span class="fl">1000000</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##  [1] 1e+00 1e+01 2e+01 5e+01 1e+02 2e+02 5e+02 1e+03 2e+03 5e+03 1e+04 2e+04</span></span>
<span><span class="co">## [13] 5e+04 1e+05 2e+05 5e+05 1e+06</span></span></code></pre>
<p>NOTE To change these parameters see <strong>Advanced
Work</strong></p>
</div>
<div class="section level4">
<h4 id="step-4---extracting-optimal-paramters-prelim">Step 4 - Extracting Optimal Paramters (Prelim)<a class="anchor" aria-label="anchor" href="#step-4---extracting-optimal-paramters-prelim"></a>
</h4>
<p>Once computed, we will extract the optimal set of parameters.</p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">optimalParam</span> <span class="op">&lt;-</span> <span class="va">optimal</span><span class="op">$</span><span class="va">Optimal</span></span>
<span><span class="va">optimalParam</span><span class="op">$</span><span class="va">OptimalParameters</span></span></code></pre></div>
<pre><code><span><span class="co">## $pearson</span></span>
<span><span class="co">##         lambda boundMolecules </span></span>
<span><span class="co">##          1e+00          1e+06 </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $spearman</span></span>
<span><span class="co">##         lambda boundMolecules </span></span>
<span><span class="co">##            0.5            1.0 </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $kendall</span></span>
<span><span class="co">##         lambda boundMolecules </span></span>
<span><span class="co">##            0.5            1.0 </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $KsDist</span></span>
<span><span class="co">##         lambda boundMolecules </span></span>
<span><span class="co">##           4.75         200.00 </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $geometric</span></span>
<span><span class="co">##         lambda boundMolecules </span></span>
<span><span class="co">##              5            100 </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $precision</span></span>
<span><span class="co">##         lambda boundMolecules </span></span>
<span><span class="co">##            0.5            1.0 </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $recall</span></span>
<span><span class="co">##         lambda boundMolecules </span></span>
<span><span class="co">##            2.5         1000.0 </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $f1</span></span>
<span><span class="co">##         lambda boundMolecules </span></span>
<span><span class="co">##           0.75           1.00 </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $accuracy</span></span>
<span><span class="co">##         lambda boundMolecules </span></span>
<span><span class="co">##           0.25        2000.00 </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $MCC</span></span>
<span><span class="co">##         lambda boundMolecules </span></span>
<span><span class="co">##            0.5            1.0 </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $AUC</span></span>
<span><span class="co">##         lambda boundMolecules </span></span>
<span><span class="co">##              2             10 </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $MSE</span></span>
<span><span class="co">##         lambda boundMolecules </span></span>
<span><span class="co">##              5             50</span></span></code></pre>
<p>We can see the opitmal set of parameters suggested by
ChIPanalyser.</p>
</div>
<div class="section level4">
<h4 id="step-5---plotting-optimal-set-of-parameters">Step 5 - Plotting Optimal Set of Parameters<a class="anchor" aria-label="anchor" href="#step-5---plotting-optimal-set-of-parameters"></a>
</h4>
<p>Despite ChIPanalyser returning suggested optimal parameters, you may
wish to visualise the optimal parameters for each paramter combination
and choose your own set of parameters. To this effect, we have
implemented an Optimal parameter plotting fucntion.</p>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Plotting Optimal heat maps</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/par.html" class="external-link">par</a></span><span class="op">(</span>oma<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span>,<span class="fl">3</span>,<span class="fl">0</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/layout.html" class="external-link">layout</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/matrix.html" class="external-link">matrix</a></span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fl">8</span>,ncol<span class="op">=</span><span class="fl">4</span>, byrow<span class="op">=</span><span class="cn">T</span><span class="op">)</span>,width<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">6</span>,<span class="fl">1.5</span>,<span class="fl">6</span>,<span class="fl">1.5</span><span class="op">)</span>,height<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>,<span class="fl">1</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="../reference/plotOptimalHeatMaps.html">plotOptimalHeatMaps</a></span><span class="op">(</span><span class="va">optimalParam</span>,layout<span class="op">=</span><span class="cn">FALSE</span><span class="op">)</span></span></code></pre></div>
<p><img src="ChIPanalyser_files/figure-html/unnamed-chunk-8-1.png" width="960"><img src="ChIPanalyser_files/figure-html/unnamed-chunk-8-2.png" width="960"><img src="ChIPanalyser_files/figure-html/unnamed-chunk-8-3.png" width="960"></p>
</div>
<div class="section level4">
<h4 id="step-6---extracting-optimal-set-of-parameters-with-associated-data">Step 6 - Extracting Optimal Set of Parameters with associated
data<a class="anchor" aria-label="anchor" href="#step-6---extracting-optimal-set-of-parameters-with-associated-data"></a>
</h4>
<p>Once satisfied with your choice of optimal parameters, you can
extracted the data associated to those parameters. You can select more
that one parameter however the values assigned to each argument must be
one of the values used for computing the optimal set of parameters.</p>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">optimalParam</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/searchSites.html">searchSites</a></span><span class="op">(</span><span class="va">optimal</span>,lambdaPWM <span class="op">=</span> <span class="fl">1.25</span>,BoundMolecules <span class="op">=</span> <span class="fl">10000</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="step-7---plotting-chip_seq-like-profiles">Step 7 - Plotting ChIP_seq like profiles<a class="anchor" aria-label="anchor" href="#step-7---plotting-chip_seq-like-profiles"></a>
</h4>
<p>The final step is to plot the computed predicted profiles. We provide
a plotting function to produce predicted profile plots.</p>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plotOccupancyProfile.html">plotOccupancyProfile</a></span><span class="op">(</span>predictedProfile <span class="op">=</span> <span class="va">optimalParam</span><span class="op">$</span><span class="va">ChIPProfiles</span>,</span>
<span>                     ChIPScore <span class="op">=</span> <span class="va">chip</span>,</span>
<span>                     chromatinState <span class="op">=</span> <span class="va">Access</span>,</span>
<span>                     occupancy <span class="op">=</span> <span class="va">optimalParam</span><span class="op">$</span><span class="va">Occupancy</span>,</span>
<span>                     goodnessOfFit <span class="op">=</span> <span class="va">optimalParam</span><span class="op">$</span><span class="va">goodnessOfFit</span>,</span>
<span>                     geneRef <span class="op">=</span> <span class="va">geneRef</span><span class="op">)</span></span></code></pre></div>
<p><img src="ChIPanalyser_files/figure-html/unnamed-chunk-10-1.png" width="1152"><img src="ChIPanalyser_files/figure-html/unnamed-chunk-10-2.png" width="1152"><img src="ChIPanalyser_files/figure-html/unnamed-chunk-10-3.png" width="1152"><img src="ChIPanalyser_files/figure-html/unnamed-chunk-10-4.png" width="1152"></p>
</div>
</div>
<div class="section level3">
<h3 id="advanced-work">Advanced Work<a class="anchor" aria-label="anchor" href="#advanced-work"></a>
</h3>
<p>In this section we will describe a more in depth work flow. This will
include parameter tweakin, annexe functions and some insights into the
outputs of each functions.</p>
<div class="section level4">
<h4 id="step-1---parameter-set-up">Step 1 - Parameter Set Up<a class="anchor" aria-label="anchor" href="#step-1---parameter-set-up"></a>
</h4>
<p>Some parameters can be changed between each step of the pipeline.
These parameters will enable you to tweak and improve the quality of
your analysis.</p>
<p>There are many parameters to choose from. These parameters already
have default value assigned to them.</p>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Suggested Parameters to start with.</span></span>
<span><span class="fu"><a href="../reference/parameterOptions.html">parameterOptions</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ </span></span>
<span><span class="co">##                         parameterOptions </span></span>
<span><span class="co">## _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ </span></span>
<span><span class="co">##  processingChIP options </span></span>
<span><span class="co">##  </span></span>
<span><span class="co">##  chipMean:                                                  200 </span></span>
<span><span class="co">##  chipSd:                                                    200 </span></span>
<span><span class="co">##  chipSmooth:                                                250 </span></span>
<span><span class="co">##  lociWidth:                                               20000 </span></span>
<span><span class="co">##  noiseFilter:                                              zero </span></span>
<span><span class="co">##  </span></span>
<span><span class="co">## _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ </span></span>
<span><span class="co">##  processingChIP options Updated </span></span>
<span><span class="co">##  </span></span>
<span><span class="co">##  maxSignal:                                                   1 </span></span>
<span><span class="co">##  backgroundSignal:                                            0 </span></span>
<span><span class="co">##  </span></span>
<span><span class="co">## _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ </span></span>
<span><span class="co">##  Position Weight Matrix Options </span></span>
<span><span class="co">##  </span></span>
<span><span class="co">##  naturalLog:                                               TRUE </span></span>
<span><span class="co">##  noOfSites:                                                 all </span></span>
<span><span class="co">##  PWMpseudocount:                                              1 </span></span>
<span><span class="co">##  </span></span>
<span><span class="co">## _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ </span></span>
<span><span class="co">##  Genome Wide Score options </span></span>
<span><span class="co">##  </span></span>
<span><span class="co">##  strandRule:                                                max </span></span>
<span><span class="co">##  whichstrand:                                                +- </span></span>
<span><span class="co">##  lambdaPWM:                                                   1 </span></span>
<span><span class="co">##  </span></span>
<span><span class="co">## _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ </span></span>
<span><span class="co">##  PWM Scores above Threshold options </span></span>
<span><span class="co">##  </span></span>
<span><span class="co">##  strandRule:                                                max </span></span>
<span><span class="co">##  whichstrand:                                                +- </span></span>
<span><span class="co">##  </span></span>
<span><span class="co">## _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ </span></span>
<span><span class="co">##  Occupancy options </span></span>
<span><span class="co">##  </span></span>
<span><span class="co">##  Ploidy:                                                      2 </span></span>
<span><span class="co">##  lambdaPWM:                                                   1 </span></span>
<span><span class="co">##  boundMolecules:                                           1000 </span></span>
<span><span class="co">##  maxSignal:                                                   1 </span></span>
<span><span class="co">##  backgroundSignal:                                            0 </span></span>
<span><span class="co">##  </span></span>
<span><span class="co">## _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ </span></span>
<span><span class="co">##  ChIP Profile options </span></span>
<span><span class="co">##  </span></span>
<span><span class="co">##  chipMean:                                                  200 </span></span>
<span><span class="co">##  chipSd:                                                    200 </span></span>
<span><span class="co">##  chipSmooth:                                                250 </span></span>
<span><span class="co">##  stepSize:                                                   10 </span></span>
<span><span class="co">##  </span></span>
<span><span class="co">## _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _</span></span></code></pre>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Changing parameters</span></span>
<span><span class="va">PO</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/parameterOptions.html">parameterOptions</a></span><span class="op">(</span>noiseFilter<span class="op">=</span><span class="st">"sigmoid"</span>,chipSd<span class="op">=</span><span class="fl">150</span>,chipMean<span class="op">=</span><span class="fl">150</span>,lociWidth<span class="op">=</span><span class="fl">30000</span><span class="op">)</span></span></code></pre></div>
<p>NOTE If you wish to do so, you can change all your parameters at this
step. These parameters will be parsed through the different steps of the
pipeline as long as you parse this object to each step of the
analysis.</p>
</div>
<div class="section level4">
<h4 id="step-2---extracting-normalised-chip-scores-">Step 2 - Extracting Normalised ChIP scores.<a class="anchor" aria-label="anchor" href="#step-2---extracting-normalised-chip-scores-"></a>
</h4>
<p>In some case you will not have a pre-determined idea of which loci
you wish to look at. The <code>processingChIP</code> function offers a
few possibilities to work around this issue.</p>
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Top 50 loci based on ChIP score</span></span>
<span><span class="fu"><a href="../reference/processingChIP.html">processingChIP</a></span><span class="op">(</span>profile <span class="op">=</span> <span class="st">"/path/to/ChIP"</span>,</span>
<span>               loci <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>               reduce <span class="op">=</span> <span class="fl">50</span>,</span>
<span>               parameterOptions <span class="op">=</span> <span class="va">PO</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Top 50 loci ALSO containing peaks</span></span>
<span><span class="fu"><a href="../reference/processingChIP.html">processingChIP</a></span><span class="op">(</span>profile <span class="op">=</span> <span class="st">"/path/to/ChIP"</span>,</span>
<span>               loci <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>               reduce <span class="op">=</span> <span class="fl">50</span>,</span>
<span>               peaks <span class="op">=</span> <span class="st">"/path/to/peaks"</span>,</span>
<span>               parameterOptions<span class="op">=</span><span class="va">PO</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Top 50 loci containing BOTH peaks and Accessible DNA</span></span>
<span><span class="fu"><a href="../reference/processingChIP.html">processingChIP</a></span><span class="op">(</span>profile <span class="op">=</span> <span class="st">"/path/to/ChIP"</span>,</span>
<span>               loci <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>               reduce <span class="op">=</span> <span class="fl">50</span>,</span>
<span>               peaks <span class="op">=</span> <span class="st">"/path/to/peaks"</span>,</span>
<span>               chromatinState <span class="op">=</span> <span class="st">"/path/to/chromatinState"</span>,</span>
<span>               parameterOptions <span class="op">=</span> <span class="va">PO</span><span class="op">)</span></span></code></pre></div>
<p>Loci will be computed internally based on the ChIP score provided.
ChIP Scores will be tilled into bins of width equals to the value
assigned to the <code>lociWidth</code> argument in the
<code>parameterOptions</code> object (see above). The default loci width
is set at 20 kbp. Top regions are selected based on ordered ChIP
scores.</p>
<p><strong>Most genomic formats are supported by ChIPanalyser (wig, bed,
bedGraph, bigWig, bigBed). The “path/to/file” may also be a GRanges
object. </strong></p>
<p><code>processingChIP</code> function returns extracted/Normalised
ChIP scores (list of numeric vectors), the loci of inetrest (GRanges),
and associated paramters that have been extracted (such as maxSignal and
backgroundSignal). The loci are the top <em>n</em> regions as selected
by the reduce argument. Using the <code><a href="../reference/loci.html">loci()</a></code> accessor applied
on a ChIPScore object will return a GRanges of selected loci. The
<code><a href="../reference/scores.html">scores()</a></code> accessors applied on a <code>ChIPScore</code>
object will return the normalised scores associated to each Locus.</p>
<p>NOTE This function also supports multi core computing.</p>
</div>
<div class="section level4">
<h4 id="step-3---position-weight-matrix-and-associated-paramters">Step 3 - Position Weight Matrix and Associated Paramters<a class="anchor" aria-label="anchor" href="#step-3---position-weight-matrix-and-associated-paramters"></a>
</h4>
<p>Computing the PWM from PFMs can be tweaked by using some additional
parameters. PWMs depend on Base Pair Frequency in the genome of
interest. Either you can provide a vector containing the base pair
frequency (A C T G in order) or the <code>genomicProfiles</code> object
will compute it internally if you provide a BSgenome/DNAStringSet.</p>
<div class="sourceCode" id="cb27"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/str.html" class="external-link">str</a></span><span class="op">(</span><span class="fu"><a href="../reference/genomicProfiles.html">genomicProfiles</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Formal class 'genomicProfiles' [package "ChIPanalyser"] with 31 slots</span></span>
<span><span class="co">##   ..@ PWM               : num[0 , 0 ] </span></span>
<span><span class="co">##   ..@ PFM               : num[0 , 0 ] </span></span>
<span><span class="co">##   ..@ PFMFormat         : chr "raw"</span></span>
<span><span class="co">##   ..@ BPFrequency       : num [1:4] 0.25 0.25 0.25 0.25</span></span>
<span><span class="co">##   ..@ minPWMScore       : logi(0) </span></span>
<span><span class="co">##   ..@ maxPWMScore       : logi(0) </span></span>
<span><span class="co">##   ..@ profiles          :Formal class 'CompressedGRangesList' [package "GenomicRanges"] with 5 slots</span></span>
<span><span class="co">##   .. .. ..@ unlistData     :Formal class 'GRanges' [package "GenomicRanges"] with 7 slots</span></span>
<span><span class="co">##   .. .. .. .. ..@ seqnames       :Formal class 'Rle' [package "S4Vectors"] with 4 slots</span></span>
<span><span class="co">##   .. .. .. .. .. .. ..@ values         : Factor w/ 0 levels: </span></span>
<span><span class="co">##   .. .. .. .. .. .. ..@ lengths        : int(0) </span></span>
<span><span class="co">##   .. .. .. .. .. .. ..@ elementMetadata: NULL</span></span>
<span><span class="co">##   .. .. .. .. .. .. ..@ metadata       : list()</span></span>
<span><span class="co">##   .. .. .. .. ..@ ranges         :Formal class 'IRanges' [package "IRanges"] with 6 slots</span></span>
<span><span class="co">##   .. .. .. .. .. .. ..@ start          : int(0) </span></span>
<span><span class="co">##   .. .. .. .. .. .. ..@ width          : int(0) </span></span>
<span><span class="co">##   .. .. .. .. .. .. ..@ NAMES          : NULL</span></span>
<span><span class="co">##   .. .. .. .. .. .. ..@ elementType    : chr "ANY"</span></span>
<span><span class="co">##   .. .. .. .. .. .. ..@ elementMetadata: NULL</span></span>
<span><span class="co">##   .. .. .. .. .. .. ..@ metadata       : list()</span></span>
<span><span class="co">##   .. .. .. .. ..@ strand         :Formal class 'Rle' [package "S4Vectors"] with 4 slots</span></span>
<span><span class="co">##   .. .. .. .. .. .. ..@ values         : Factor w/ 3 levels "+","-","*": </span></span>
<span><span class="co">##   .. .. .. .. .. .. ..@ lengths        : int(0) </span></span>
<span><span class="co">##   .. .. .. .. .. .. ..@ elementMetadata: NULL</span></span>
<span><span class="co">##   .. .. .. .. .. .. ..@ metadata       : list()</span></span>
<span><span class="co">##   .. .. .. .. ..@ seqinfo        :Formal class 'Seqinfo' [package "GenomeInfoDb"] with 4 slots</span></span>
<span><span class="co">##   .. .. .. .. .. .. ..@ seqnames   : chr(0) </span></span>
<span><span class="co">##   .. .. .. .. .. .. ..@ seqlengths : int(0) </span></span>
<span><span class="co">##   .. .. .. .. .. .. ..@ is_circular: logi(0) </span></span>
<span><span class="co">##   .. .. .. .. .. .. ..@ genome     : chr(0) </span></span>
<span><span class="co">##   .. .. .. .. ..@ elementMetadata:Formal class 'DFrame' [package "S4Vectors"] with 6 slots</span></span>
<span><span class="co">##   .. .. .. .. .. .. ..@ rownames       : NULL</span></span>
<span><span class="co">##   .. .. .. .. .. .. ..@ nrows          : int 0</span></span>
<span><span class="co">##   .. .. .. .. .. .. ..@ elementType    : chr "ANY"</span></span>
<span><span class="co">##   .. .. .. .. .. .. ..@ elementMetadata: NULL</span></span>
<span><span class="co">##   .. .. .. .. .. .. ..@ metadata       : list()</span></span>
<span><span class="co">##   .. .. .. .. .. .. ..@ listData       : Named list()</span></span>
<span><span class="co">##   .. .. .. .. ..@ elementType    : chr "ANY"</span></span>
<span><span class="co">##   .. .. .. .. ..@ metadata       : list()</span></span>
<span><span class="co">##   .. .. ..@ elementMetadata:Formal class 'DFrame' [package "S4Vectors"] with 6 slots</span></span>
<span><span class="co">##   .. .. .. .. ..@ rownames       : NULL</span></span>
<span><span class="co">##   .. .. .. .. ..@ nrows          : int 0</span></span>
<span><span class="co">##   .. .. .. .. ..@ elementType    : chr "ANY"</span></span>
<span><span class="co">##   .. .. .. .. ..@ elementMetadata: NULL</span></span>
<span><span class="co">##   .. .. .. .. ..@ metadata       : list()</span></span>
<span><span class="co">##   .. .. .. .. ..@ listData       : Named list()</span></span>
<span><span class="co">##   .. .. ..@ elementType    : chr "GRanges"</span></span>
<span><span class="co">##   .. .. ..@ metadata       : list()</span></span>
<span><span class="co">##   .. .. ..@ partitioning   :Formal class 'PartitioningByEnd' [package "IRanges"] with 5 slots</span></span>
<span><span class="co">##   .. .. .. .. ..@ end            : int(0) </span></span>
<span><span class="co">##   .. .. .. .. ..@ NAMES          : NULL</span></span>
<span><span class="co">##   .. .. .. .. ..@ elementType    : chr "ANY"</span></span>
<span><span class="co">##   .. .. .. .. ..@ elementMetadata: NULL</span></span>
<span><span class="co">##   .. .. .. .. ..@ metadata       : list()</span></span>
<span><span class="co">##   ..@ DNASequenceLength : logi(0) </span></span>
<span><span class="co">##   ..@ averageExpPWMScore: logi(0) </span></span>
<span><span class="co">##   ..@ ZeroBackground    : logi(0) </span></span>
<span><span class="co">##   ..@ drop              : logi(0) </span></span>
<span><span class="co">##   ..@ tags              : chr "empty"</span></span>
<span><span class="co">##   ..@ ploidy            : num 2</span></span>
<span><span class="co">##   ..@ boundMolecules    : num 1000</span></span>
<span><span class="co">##   ..@ backgroundSignal  : num 0</span></span>
<span><span class="co">##   ..@ maxSignal         : num 1</span></span>
<span><span class="co">##   ..@ lociWidth         : num 20000</span></span>
<span><span class="co">##   ..@ chipMean          : num 200</span></span>
<span><span class="co">##   ..@ chipSd            : num 200</span></span>
<span><span class="co">##   ..@ chipSmooth        : num 250</span></span>
<span><span class="co">##   ..@ stepSize          : num 10</span></span>
<span><span class="co">##   ..@ removeBackground  : num 0</span></span>
<span><span class="co">##   ..@ noiseFilter       : chr "zero"</span></span>
<span><span class="co">##   ..@ PWMThreshold      : num 0.7</span></span>
<span><span class="co">##   ..@ strandRule        : chr "max"</span></span>
<span><span class="co">##   ..@ whichstrand       : chr "+-"</span></span>
<span><span class="co">##   ..@ lambdaPWM         : num 1</span></span>
<span><span class="co">##   ..@ naturalLog        : logi TRUE</span></span>
<span><span class="co">##   ..@ noOfSites         : chr "all"</span></span>
<span><span class="co">##   ..@ PWMpseudocount    : num 1</span></span>
<span><span class="co">##   ..@ paramTag          : chr "empty"</span></span></code></pre>
<div class="sourceCode" id="cb29"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">GP</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/genomicProfiles.html">genomicProfiles</a></span><span class="op">(</span>PFM<span class="op">=</span><span class="va">PFM</span>, PFMFormat<span class="op">=</span><span class="st">"JASPAR"</span>, BPFrequency<span class="op">=</span><span class="va">DNASequenceSet</span><span class="op">)</span></span>
<span><span class="va">GP</span></span></code></pre></div>
<pre><code><span><span class="co">## _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ </span></span>
<span><span class="co">##  </span></span>
<span><span class="co">##                         genomicProfiles object: Position Weight Matrix </span></span>
<span><span class="co">## _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ </span></span>
<span><span class="co">##  </span></span>
<span><span class="co">## Position Weight Matrix (PWM): </span></span>
<span><span class="co">##  </span></span>
<span><span class="co">##          [,1]        [,2]        [,3]       [,4]       [,5]       [,6]</span></span>
<span><span class="co">## A -0.43767288 -0.18659373  0.08313105  0.4815961  0.5392208  0.6417377</span></span>
<span><span class="co">## C  0.04045561  0.53988434  0.01137399  0.4833483 -0.8893224 -0.9491829</span></span>
<span><span class="co">## G  0.30241584  0.01243766  0.44766827 -0.1011918 -0.2204289  0.1913679</span></span>
<span><span class="co">## T  0.06672312 -0.44259797 -0.70094631 -8.2905435 -0.1563618 -0.9380145</span></span>
<span><span class="co">##         [,7]       [,8]      [,9]     [,10]     [,11]     [,12]      [,13]</span></span>
<span><span class="co">## A  0.2914731 -0.4390137  1.218068 -8.290544 -8.290544 -2.705222  1.1395868</span></span>
<span><span class="co">## C  0.7876035 -0.2766189 -8.290544 -8.290544  1.559771 -8.290544 -8.2905435</span></span>
<span><span class="co">## G -0.3372422 -8.2905435 -2.382983 -8.290544 -8.290544  1.541324 -0.8100529</span></span>
<span><span class="co">## T -8.2905435  0.8129614 -8.290544  1.238027 -8.290544 -8.290544 -8.2905435</span></span>
<span><span class="co">##        [,14]      [,15]</span></span>
<span><span class="co">## A -0.9929866  0.6261359</span></span>
<span><span class="co">## C -1.5935410 -8.2905435</span></span>
<span><span class="co">## G -8.2905435 -0.0597464</span></span>
<span><span class="co">## T  1.0754476 -0.1099011</span></span>
<span><span class="co">## _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ </span></span>
<span><span class="co">##  </span></span>
<span><span class="co">## Base pairs frequency for PWM weighting : </span></span>
<span><span class="co">##  </span></span>
<span><span class="co">##         A         C         G         T </span></span>
<span><span class="co">## 0.2900342 0.2101426 0.2099192 0.2899039 </span></span>
<span><span class="co">## _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ </span></span>
<span><span class="co">##  </span></span>
<span><span class="co">## PWM built from (PFM - Format  JASPAR ) : </span></span>
<span><span class="co">##  </span></span>
<span><span class="co">##   [,1] [,2] [,3] [,4] [,5] [,6] [,7] [,8] [,9] [,10] [,11] [,12] [,13] [,14]</span></span>
<span><span class="co">## A  746  959 1256 1871 1982 2196 1547  745 3908     0     0    77  3613   428</span></span>
<span><span class="co">## C  872 1437  847 1358  344  324 1841  635    0     0  3985     0     0   170</span></span>
<span><span class="co">## G 1132  847 1309  756  671 1013  597    0   77     0     0  3908   372     0</span></span>
<span><span class="co">## T 1235  742  573    0  988  452    0 2605    0  3985     0     0     0  3387</span></span>
<span><span class="co">##   [,15]</span></span>
<span><span class="co">## A  2162</span></span>
<span><span class="co">## C     0</span></span>
<span><span class="co">## G   788</span></span>
<span><span class="co">## T  1035</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##  _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _</span></span></code></pre>
<p>The <code>genomicProfiles</code> object also contains all parameters
described in <code>parameterOptions</code>. This makes the parsing of
parameters straight forward between each step of the analysis pipeline.
The <code>genomicProfiles</code> object will be updated internally as
you provided More parameters. This object will also be the main object
that you will parse through each step of the analysis. There are a few
different ways to add new paramters:</p>
<div class="sourceCode" id="cb31"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Parsing pre computed parameters (processingChIP function)</span></span>
<span><span class="va">GP</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/genomicProfiles.html">genomicProfiles</a></span><span class="op">(</span>PFM <span class="op">=</span> <span class="va">PFM</span>, </span>
<span>                     PFMFormat <span class="op">=</span> <span class="st">"JASPAR"</span>,</span>
<span>                     BPFrequency <span class="op">=</span> <span class="va">DNASequenceSet</span>,</span>
<span>                     ChIPScore <span class="op">=</span> <span class="va">ChIPScore</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Parsing pre assigned function (parameterOptions)</span></span>
<span><span class="va">parameterOptions</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/parameterOptions.html">parameterOptions</a></span><span class="op">(</span>lambdaPWM <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>,<span class="fl">2</span>,<span class="fl">3</span><span class="op">)</span>,</span>
<span>                                     boundMolecules <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">5</span>,<span class="fl">50</span>,<span class="fl">500</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">GP</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/genomicProfiles.html">genomicProfiles</a></span><span class="op">(</span>PFM <span class="op">=</span> <span class="va">PFM</span>, </span>
<span>                     PFMFormat <span class="op">=</span> <span class="st">"JASPAR"</span>, </span>
<span>                     BPFrequency <span class="op">=</span> <span class="va">DNASequenceSet</span>,</span>
<span>                     parameterOptions <span class="op">=</span> <span class="va">parameterOptions</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Direct parameter assignement</span></span>
<span></span>
<span><span class="va">GP</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/genomicProfiles.html">genomicProfiles</a></span><span class="op">(</span>PFM <span class="op">=</span> <span class="va">PFM</span>, </span>
<span>                     PFMFormat <span class="op">=</span> <span class="st">"JASPAR"</span>, </span>
<span>                     BPFrequency <span class="op">=</span> <span class="va">DNASequenceSet</span>, </span>
<span>                     lambdaPWM <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>,<span class="fl">2</span>,<span class="fl">3</span><span class="op">)</span>,</span>
<span>                     boundMolecules <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">4</span>,<span class="fl">500</span>,<span class="fl">8000</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p>NOTE <code>parameterOptions</code> object can be parsed to any
function of the analysis pipeline if parameters need to be changed along
the way.</p>
</div>
<div class="section level4">
<h4 id="step-4---computing-optimal-set-of-parameters">Step 4 - Computing Optimal Set of Parameters<a class="anchor" aria-label="anchor" href="#step-4---computing-optimal-set-of-parameters"></a>
</h4>
<p>The optimal set of parameters can be computed on custom set of values
for <span class="math inline">\(N\)</span> and <span class="math inline">\(\lambda\)</span>. As described above, there a few
ways to modify parameter Options. If you were to assign more than two
values to both of these slots, these new values will be used as “Optimal
Parameters Combinations”. NOTE <code>parameterOptions</code> is
inherited by <code>genomicProfiles</code> hence why you can also assign
those paramters to the <code>genomicProfiles</code> contructor.</p>
<p>The <code>computeOptimal</code> function offers a few more options
that we will briefly describe here.</p>
<div class="sourceCode" id="cb32"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Setting custom parameters</span></span>
<span><span class="va">OP</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/parameterOptions.html">parameterOptions</a></span><span class="op">(</span>lambdaPWM <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq</a></span><span class="op">(</span><span class="fl">1</span>,<span class="fl">10</span>,by<span class="op">=</span><span class="fl">0.5</span><span class="op">)</span>, </span>
<span>                      boundMolecules <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq</a></span><span class="op">(</span><span class="fl">1</span>,<span class="fl">100000</span>, length.out<span class="op">=</span><span class="fl">20</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Computing ONLY Optimal Parameters and MSE as goodness Of Fit metric</span></span>
<span><span class="va">optimal</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/computeOptimal.html">computeOptimal</a></span><span class="op">(</span>genomicProfiles <span class="op">=</span> <span class="va">GP</span>,</span>
<span>                         DNASequenceSet <span class="op">=</span> <span class="va">DNASequenceSet</span>,</span>
<span>                         ChIPScore <span class="op">=</span> <span class="va">chip</span>,</span>
<span>                         chromatinState <span class="op">=</span> <span class="va">Access</span>,</span>
<span>                         parameterOptions <span class="op">=</span> <span class="va">OP</span>,</span>
<span>                         optimalMethod <span class="op">=</span> <span class="st">"MSE"</span>,</span>
<span>                         returnAll <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span></span>
<span><span class="co">### Computing ONLY Optimal Parameters and using Rank slection method</span></span>
<span><span class="va">optimal</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/computeOptimal.html">computeOptimal</a></span><span class="op">(</span>genomicProfiles <span class="op">=</span> <span class="va">GP</span>,</span>
<span>                         DNASequenceSet <span class="op">=</span> <span class="va">DNASequenceSet</span>,</span>
<span>                         ChIPScore <span class="op">=</span> <span class="va">chip</span>,</span>
<span>                         chromatinState <span class="op">=</span> <span class="va">Access</span>,</span>
<span>                         parameterOptions <span class="op">=</span> <span class="va">OP</span>,</span>
<span>                         optimalMethod <span class="op">=</span> <span class="st">"all"</span>,</span>
<span>                         rank<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<p>When the <code>returnAll</code> argument is set to <em>FALSE</em>,
only the optimal parameters Will be return. No internal Data will be
returned.</p>
<p>Optimal Parameters are determined by selecting the best perfomring
combination of paremeters. The goodness of fit score for each
combination is averaged over all regions considered. When the
<code>rank</code> argument is set to <em>TRUE</em>, the optimal
parameters will be based on which combination of parameters showed the
best performance for each regions individually. Parameter combinations
are ranked based on how many individual regions performed best with that
specific set of parameters.</p>
<p>Finally, <code>optimalMethod</code> argument will enbale you to
selected the goodness Of Fit method you wish to use.</p>
</div>
<div class="section level4">
<h4 id="step-5---extracting-and-plotting-optimal-parameters">Step 5 - Extracting and Plotting Optimal Parameters<a class="anchor" aria-label="anchor" href="#step-5---extracting-and-plotting-optimal-parameters"></a>
</h4>
<p>Now that you have selected custom parameters, you will want to plot
the associated heat maps.</p>
<div class="sourceCode" id="cb33"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Extracted Optimal Parameters</span></span>
<span><span class="va">optimalParam</span> <span class="op">&lt;-</span> <span class="va">optimal</span><span class="op">$</span><span class="va">Optimal</span></span>
<span></span>
<span><span class="co">## Plotting heat maps</span></span>
<span><span class="fu"><a href="../reference/plotOptimalHeatMaps.html">plotOptimalHeatMaps</a></span><span class="op">(</span><span class="va">optimalParam</span>,overlay<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<p>It is possible to plot an overlay of the optimal set of paramters of
all goodness Of Fit methods. Using the <code>overlay</code> argument in
the plotting function will plot overlay the top 10% of optimal
parameters as selected by each Goodness of fit metric.</p>
</div>
<div class="section level4">
<h4 id="step-6---computing-individual-parameter-combinations">Step 6 - Computing individual parameter combinations<a class="anchor" aria-label="anchor" href="#step-6---computing-individual-parameter-combinations"></a>
</h4>
<p>Let’s imagine that when looking at the optimal parameter heat maps,
you would like to run a combination of parameters that is not in the
ones that had been provided but you do not want to re-compute optimal
parameters. Or Let us imagine that you have already an estimate of
number of bound molecules. ChIPanalyser provides functions that will
enable you to run the piple line on individual parameter combinations.
The steps are described as following:</p>
<div class="sourceCode" id="cb34"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Creating genomic Profiles object with PFMs and associated parameters</span></span>
<span><span class="va">GP</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/genomicProfiles.html">genomicProfiles</a></span><span class="op">(</span>PFM <span class="op">=</span> <span class="va">PFM</span>,</span>
<span>                      PFMFormat <span class="op">=</span> <span class="st">"JASPAR"</span>,</span>
<span>                      BPFrequency <span class="op">=</span> <span class="va">DNASequenceSet</span>,</span>
<span>                      lambdaPWM <span class="op">=</span> <span class="fl">1</span>, </span>
<span>                      boundMolecules <span class="op">=</span> <span class="fl">58794</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Computing Genome Wide Score required</span></span>
<span><span class="va">GW</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/computeGenomeWideScores.html">computeGenomeWideScores</a></span><span class="op">(</span>genomicProfiles <span class="op">=</span> <span class="va">GP</span>,</span>
<span>                              DNASequenceSet <span class="op">=</span> <span class="va">DNASequenceSet</span>,</span>
<span>                              chromatinState <span class="op">=</span> <span class="va">Access</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Extracting genome wide scores</span></span></code></pre>
<pre><code><span><span class="co">## Considering Chromatin State ~ Both strands</span></span></code></pre>
<pre><code><span><span class="co">## Computing Mean waiting time</span></span></code></pre>
<div class="sourceCode" id="cb38"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">GW</span></span></code></pre></div>
<pre><code><span><span class="co">## _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ </span></span>
<span><span class="co">##  </span></span>
<span><span class="co">##                         genomicProfiles object: Accessible Chromatin Score </span></span>
<span><span class="co">## _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ </span></span>
<span><span class="co">##  </span></span>
<span><span class="co">##  Position Weight Matrix (PWM): </span></span>
<span><span class="co">##  </span></span>
<span><span class="co">##          [,1]        [,2]        [,3]       [,4]       [,5]       [,6]</span></span>
<span><span class="co">## A -0.43767288 -0.18659373  0.08313105  0.4815961  0.5392208  0.6417377</span></span>
<span><span class="co">## C  0.04045561  0.53988434  0.01137399  0.4833483 -0.8893224 -0.9491829</span></span>
<span><span class="co">## G  0.30241584  0.01243766  0.44766827 -0.1011918 -0.2204289  0.1913679</span></span>
<span><span class="co">## T  0.06672312 -0.44259797 -0.70094631 -8.2905435 -0.1563618 -0.9380145</span></span>
<span><span class="co">##         [,7]       [,8]      [,9]     [,10]     [,11]     [,12]      [,13]</span></span>
<span><span class="co">## A  0.2914731 -0.4390137  1.218068 -8.290544 -8.290544 -2.705222  1.1395868</span></span>
<span><span class="co">## C  0.7876035 -0.2766189 -8.290544 -8.290544  1.559771 -8.290544 -8.2905435</span></span>
<span><span class="co">## G -0.3372422 -8.2905435 -2.382983 -8.290544 -8.290544  1.541324 -0.8100529</span></span>
<span><span class="co">## T -8.2905435  0.8129614 -8.290544  1.238027 -8.290544 -8.290544 -8.2905435</span></span>
<span><span class="co">##        [,14]      [,15]</span></span>
<span><span class="co">## A -0.9929866  0.6261359</span></span>
<span><span class="co">## C -1.5935410 -8.2905435</span></span>
<span><span class="co">## G -8.2905435 -0.0597464</span></span>
<span><span class="co">## T  1.0754476 -0.1099011</span></span>
<span><span class="co">## _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ </span></span>
<span><span class="co">##  </span></span>
<span><span class="co">##  maxPWMScore:                                11.1310200679306</span></span>
<span><span class="co">##  minPWMScore:                               -59.5942907981563</span></span>
<span><span class="co">##  averageExpPWMScore:             8.29141420231848 (lambda = 1)</span></span>
<span><span class="co">##  DNASequenceLength:                                     10471</span></span>
<span><span class="co">## _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ </span></span>
<span><span class="co">## </span></span></code></pre>
<div class="sourceCode" id="cb40"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Computing PWM score above threshold</span></span>
<span><span class="va">pwm</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/computePWMScore.html">computePWMScore</a></span><span class="op">(</span>genomicProfiles <span class="op">=</span> <span class="va">GW</span>,</span>
<span>                       DNASequenceSet <span class="op">=</span> <span class="va">DNASequenceSet</span>,</span>
<span>                       loci <span class="op">=</span> <span class="va">chip</span>,</span>
<span>                       chromatinState <span class="op">=</span> <span class="va">Access</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## PWM Scores Extraction</span></span></code></pre>
<div class="sourceCode" id="cb42"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">pwm</span></span></code></pre></div>
<pre><code><span><span class="co">## _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ </span></span>
<span><span class="co">##  </span></span>
<span><span class="co">##                         genomicProfiles object: PWM scores above Threshold </span></span>
<span><span class="co">## _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ </span></span>
<span><span class="co">##  </span></span>
<span><span class="co">##  Scores above Threshold in locus:  </span></span>
<span><span class="co">## GRanges object with 169 ranges and 2 metadata columns:</span></span>
<span><span class="co">##                            seqnames            ranges strand |  PWMScore</span></span>
<span><span class="co">##                               &lt;Rle&gt;         &lt;IRanges&gt;  &lt;Rle&gt; | &lt;numeric&gt;</span></span>
<span><span class="co">##   chr3L:20300001..20320000    chr3L 20300005-20300019      - | -0.260743</span></span>
<span><span class="co">##   chr3L:20300001..20320000    chr3L 20300015-20300029      + | -3.581791</span></span>
<span><span class="co">##   chr3L:20300001..20320000    chr3L 20300022-20300036      - |  2.266402</span></span>
<span><span class="co">##   chr3L:20300001..20320000    chr3L 20300066-20300080      + | -7.941449</span></span>
<span><span class="co">##   chr3L:20300001..20320000    chr3L 20300293-20300307      + | -6.443234</span></span>
<span><span class="co">##                        ...      ...               ...    ... .       ...</span></span>
<span><span class="co">##   chr3L:20300001..20320000    chr3L 20318607-20318621      - |  -9.10951</span></span>
<span><span class="co">##   chr3L:20300001..20320000    chr3L 20318620-20318634      + |  -3.39948</span></span>
<span><span class="co">##   chr3L:20300001..20320000    chr3L 20318676-20318690      - | -10.03808</span></span>
<span><span class="co">##   chr3L:20300001..20320000    chr3L 20318724-20318738      + |  -7.80125</span></span>
<span><span class="co">##   chr3L:20300001..20320000    chr3L 20318728-20318742      + |  -6.63695</span></span>
<span><span class="co">##                            DNAaffinity</span></span>
<span><span class="co">##                              &lt;numeric&gt;</span></span>
<span><span class="co">##   chr3L:20300001..20320000           1</span></span>
<span><span class="co">##   chr3L:20300001..20320000           1</span></span>
<span><span class="co">##   chr3L:20300001..20320000           1</span></span>
<span><span class="co">##   chr3L:20300001..20320000           1</span></span>
<span><span class="co">##   chr3L:20300001..20320000           1</span></span>
<span><span class="co">##                        ...         ...</span></span>
<span><span class="co">##   chr3L:20300001..20320000           1</span></span>
<span><span class="co">##   chr3L:20300001..20320000           1</span></span>
<span><span class="co">##   chr3L:20300001..20320000           1</span></span>
<span><span class="co">##   chr3L:20300001..20320000           1</span></span>
<span><span class="co">##   chr3L:20300001..20320000           1</span></span>
<span><span class="co">##   -------</span></span>
<span><span class="co">##   seqinfo: 1 sequence from an unspecified genome; no seqlengths</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##  3  remaining loci </span></span>
<span><span class="co">##  Locus:                chr2L:6480001..6500000 (length = 150)</span></span>
<span><span class="co">##  Locus:                chr3R:16140001..16160000 (length = 121)</span></span>
<span><span class="co">##  Locus:                chr2L:19500001..19520000 (length = 79)</span></span>
<span><span class="co">## _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ </span></span>
<span><span class="co">##  </span></span>
<span><span class="co">##                         Associated parameters </span></span>
<span><span class="co">## _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ </span></span>
<span><span class="co">##  </span></span>
<span><span class="co">##  minPWMScore:                               -59.5942907981563</span></span>
<span><span class="co">##  maxPWMScore:                                11.1310200679306</span></span>
<span><span class="co">##  lambdaPWM:                                                 1</span></span>
<span><span class="co">##  PWMThreshold:                                            0.7</span></span></code></pre>
<div class="sourceCode" id="cb44"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Computing Occupancy of sites above threshold</span></span>
<span></span>
<span><span class="va">occup</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/computeOccupancy.html">computeOccupancy</a></span><span class="op">(</span>genomicProfiles <span class="op">=</span> <span class="va">pwm</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Computing Occupancy at sites higher than threshold.</span></span></code></pre>
<div class="sourceCode" id="cb46"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">occup</span></span></code></pre></div>
<pre><code><span><span class="co">## _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ </span></span>
<span><span class="co">##  </span></span>
<span><span class="co">##                         genomicProfiles object: Occupancy at sites above Threshold </span></span>
<span><span class="co">## _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ </span></span>
<span><span class="co">##  </span></span>
<span><span class="co">##  Parameter combination:  lambda = 1 &amp; boundMolecules = 58794 </span></span>
<span><span class="co">##  </span></span>
<span><span class="co">## $`lambda = 1 &amp; boundMolecules = 58794`</span></span>
<span><span class="co">## GRangesList object of length 4:</span></span>
<span><span class="co">## $`chr3L:20300001..20320000`</span></span>
<span><span class="co">## GRanges object with 169 ranges and 3 metadata columns:</span></span>
<span><span class="co">##                            seqnames            ranges strand |  PWMScore</span></span>
<span><span class="co">##                               &lt;Rle&gt;         &lt;IRanges&gt;  &lt;Rle&gt; | &lt;numeric&gt;</span></span>
<span><span class="co">##   chr3L:20300001..20320000    chr3L 20300005-20300019      - | -0.260743</span></span>
<span><span class="co">##   chr3L:20300001..20320000    chr3L 20300015-20300029      + | -3.581791</span></span>
<span><span class="co">##   chr3L:20300001..20320000    chr3L 20300022-20300036      - |  2.266402</span></span>
<span><span class="co">##   chr3L:20300001..20320000    chr3L 20300066-20300080      + | -7.941449</span></span>
<span><span class="co">##   chr3L:20300001..20320000    chr3L 20300293-20300307      + | -6.443234</span></span>
<span><span class="co">##                        ...      ...               ...    ... .       ...</span></span>
<span><span class="co">##   chr3L:20300001..20320000    chr3L 20318607-20318621      - |  -9.10951</span></span>
<span><span class="co">##   chr3L:20300001..20320000    chr3L 20318620-20318634      + |  -3.39948</span></span>
<span><span class="co">##   chr3L:20300001..20320000    chr3L 20318676-20318690      - | -10.03808</span></span>
<span><span class="co">##   chr3L:20300001..20320000    chr3L 20318724-20318738      + |  -7.80125</span></span>
<span><span class="co">##   chr3L:20300001..20320000    chr3L 20318728-20318742      + |  -6.63695</span></span>
<span><span class="co">##                            DNAaffinity   Occupancy</span></span>
<span><span class="co">##                              &lt;numeric&gt;   &lt;numeric&gt;</span></span>
<span><span class="co">##   chr3L:20300001..20320000           1 0.206905467</span></span>
<span><span class="co">##   chr3L:20300001..20320000           1 0.009333865</span></span>
<span><span class="co">##   chr3L:20300001..20320000           1 0.765570117</span></span>
<span><span class="co">##   chr3L:20300001..20320000           1 0.000120422</span></span>
<span><span class="co">##   chr3L:20300001..20320000           1 0.000538507</span></span>
<span><span class="co">##                        ...         ...         ...</span></span>
<span><span class="co">##   chr3L:20300001..20320000           1 3.74506e-05</span></span>
<span><span class="co">##   chr3L:20300001..20320000           1 1.11796e-02</span></span>
<span><span class="co">##   chr3L:20300001..20320000           1 1.47978e-05</span></span>
<span><span class="co">##   chr3L:20300001..20320000           1 1.38543e-04</span></span>
<span><span class="co">##   chr3L:20300001..20320000           1 4.43715e-04</span></span>
<span><span class="co">##   -------</span></span>
<span><span class="co">##   seqinfo: 3 sequences from an unspecified genome; no seqlengths</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## ...</span></span>
<span><span class="co">## &lt;3 more elements&gt;</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ </span></span>
<span><span class="co">##  </span></span>
<span><span class="co">##                         Associated parameters </span></span>
<span><span class="co">## _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ </span></span>
<span><span class="co">##  </span></span>
<span><span class="co">##  lambdaPWM:                                                 1</span></span>
<span><span class="co">##  boundMolecules:                                        58794</span></span></code></pre>
<div class="sourceCode" id="cb48"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Compute ChIP seq like profiles</span></span>
<span></span>
<span><span class="va">pred</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/computeChIPProfile.html">computeChIPProfile</a></span><span class="op">(</span>genomicProfiles <span class="op">=</span> <span class="va">occup</span>,</span>
<span>                          loci <span class="op">=</span> <span class="va">chip</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Computing ChIP Profile</span></span></code></pre>
<div class="sourceCode" id="cb50"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">pred</span></span></code></pre></div>
<pre><code><span><span class="co">## _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ </span></span>
<span><span class="co">##  </span></span>
<span><span class="co">##                         genomicProfiles object: ChIP like Profiles </span></span>
<span><span class="co">## _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ </span></span>
<span><span class="co">##  </span></span>
<span><span class="co">##  Parameter combination:  lambda = 1 &amp; boundMolecules = 58794 </span></span>
<span><span class="co">##  </span></span>
<span><span class="co">## $`lambda = 1 &amp; boundMolecules = 58794`</span></span>
<span><span class="co">## GRangesList object of length 4:</span></span>
<span><span class="co">## $`chr3L:20300001..20320000`</span></span>
<span><span class="co">## GRanges object with 2000 ranges and 1 metadata column:</span></span>
<span><span class="co">##                            seqnames            ranges strand |        ChIP</span></span>
<span><span class="co">##                               &lt;Rle&gt;         &lt;IRanges&gt;  &lt;Rle&gt; |   &lt;numeric&gt;</span></span>
<span><span class="co">##   chr3L:20300001..20320000    chr3L 20300001-20300010      * |    0.237578</span></span>
<span><span class="co">##   chr3L:20300001..20320000    chr3L 20300011-20300020      * |    0.246655</span></span>
<span><span class="co">##   chr3L:20300001..20320000    chr3L 20300021-20300030      * |    0.254139</span></span>
<span><span class="co">##   chr3L:20300001..20320000    chr3L 20300031-20300040      * |    0.244935</span></span>
<span><span class="co">##   chr3L:20300001..20320000    chr3L 20300041-20300050      * |    0.234430</span></span>
<span><span class="co">##                        ...      ...               ...    ... .         ...</span></span>
<span><span class="co">##   chr3L:20300001..20320000    chr3L 20319951-20319960      * | 8.21105e-06</span></span>
<span><span class="co">##   chr3L:20300001..20320000    chr3L 20319961-20319970      * | 7.81059e-06</span></span>
<span><span class="co">##   chr3L:20300001..20320000    chr3L 20319971-20319980      * | 7.42966e-06</span></span>
<span><span class="co">##   chr3L:20300001..20320000    chr3L 20319981-20319990      * | 7.06731e-06</span></span>
<span><span class="co">##   chr3L:20300001..20320000    chr3L 20319991-20320000      * | 6.72264e-06</span></span>
<span><span class="co">##   -------</span></span>
<span><span class="co">##   seqinfo: 3 sequences from an unspecified genome; no seqlengths</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## ...</span></span>
<span><span class="co">## &lt;3 more elements&gt;</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ </span></span>
<span><span class="co">##  </span></span>
<span><span class="co">##                         Associated parameter Options </span></span>
<span><span class="co">## _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ </span></span>
<span><span class="co">##  </span></span>
<span><span class="co">##  lambdaPWM:                                                 1</span></span>
<span><span class="co">##  boundMolecules:                                        58794</span></span>
<span><span class="co">##  stepSize:                                                 10</span></span>
<span><span class="co">##  backgroundSignal:                         0.0354659475382993</span></span>
<span><span class="co">##  maxSignal:                                                 1</span></span>
<span><span class="co">##  removeBackground:                                          0</span></span>
<span><span class="co">##  chipMean:                                                200</span></span>
<span><span class="co">##  chipSd:                                                  200</span></span>
<span><span class="co">##  chipSmooth:                                              250</span></span></code></pre>
<div class="sourceCode" id="cb52"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Compute goodness Of Fit of model</span></span>
<span><span class="va">accu</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/profileAccuracyEstimate.html">profileAccuracyEstimate</a></span><span class="op">(</span>genomicProfiles <span class="op">=</span> <span class="va">pred</span>,</span>
<span>                                ChIPScore <span class="op">=</span> <span class="va">chip</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Warning in ks.test.default(predicted, locusProfile): p-value will be approximate</span></span>
<span><span class="co">## in the presence of ties</span></span>
<span></span>
<span><span class="co">## Warning in ks.test.default(predicted, locusProfile): p-value will be approximate</span></span>
<span><span class="co">## in the presence of ties</span></span>
<span></span>
<span><span class="co">## Warning in ks.test.default(predicted, locusProfile): p-value will be approximate</span></span>
<span><span class="co">## in the presence of ties</span></span>
<span></span>
<span><span class="co">## Warning in ks.test.default(predicted, locusProfile): p-value will be approximate</span></span>
<span><span class="co">## in the presence of ties</span></span></code></pre>
<div class="sourceCode" id="cb54"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">accu</span></span></code></pre></div>
<pre><code><span><span class="co">## _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ </span></span>
<span><span class="co">##  </span></span>
<span><span class="co">##                         genomicProfiles object: Goodness of Fit  </span></span>
<span><span class="co">## _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ </span></span>
<span><span class="co">##  </span></span>
<span><span class="co">##  Selected Goodness of Fit metrics:  </span></span>
<span><span class="co">##  pearson </span></span>
<span><span class="co">##  spearman </span></span>
<span><span class="co">##  kendall </span></span>
<span><span class="co">##  KsDist </span></span>
<span><span class="co">##  geometric </span></span>
<span><span class="co">##  precision </span></span>
<span><span class="co">##  recall </span></span>
<span><span class="co">##  f1 </span></span>
<span><span class="co">##  accuracy </span></span>
<span><span class="co">##  MCC </span></span>
<span><span class="co">##  AUC </span></span>
<span><span class="co">##  MSE </span></span>
<span><span class="co">## _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ </span></span>
<span><span class="co">##  </span></span>
<span><span class="co">##  1 parameter combinations for 4 loci </span></span>
<span><span class="co">## _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ </span></span>
<span><span class="co">##  </span></span>
<span><span class="co">##  Parameter combination:  lambda = 1 &amp; boundMolecules = 58794 </span></span>
<span><span class="co">##  </span></span>
<span><span class="co">##   pearsonMean  spearmanMean   kendallMean    KsDistMean geometricMean </span></span>
<span><span class="co">##    0.31991410    0.17540895    0.12094178    0.30075000    6.33325180 </span></span>
<span><span class="co">## precisionMean    recallMean        f1Mean  accuracyMean       MCCMean </span></span>
<span><span class="co">##    0.15024576    0.73827352    0.17508145    0.52158826    0.10818296 </span></span>
<span><span class="co">##       AUCMean       MSEMean </span></span>
<span><span class="co">##    0.75231385    0.02291433 </span></span>
<span><span class="co">## </span></span>
<span><span class="co">##  chr3L:20300001..20320000 </span></span>
<span><span class="co">##  chr2L:6480001..6500000 </span></span>
<span><span class="co">##  chr3R:16140001..16160000 </span></span>
<span><span class="co">##  chr2L:19500001..19520000 </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ </span></span>
<span><span class="co">## </span></span></code></pre>
<p>NOTE These functions can compute multiple parameter combinations if
needed. <code>computeOptimal</code> is essentially a combination of
these functions with a little more magic to it.</p>
<p>For more information on these functions see <strong>Parameters
Discription</strong></p>
</div>
<div class="section level4">
<h4 id="step-7---plotting-single-combination">Step 7 - Plotting Single combination<a class="anchor" aria-label="anchor" href="#step-7---plotting-single-combination"></a>
</h4>
<p>Finally, you can plot your newly computed profiles.</p>
<div class="sourceCode" id="cb56"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plotOccupancyProfile.html">plotOccupancyProfile</a></span><span class="op">(</span>predictedProfile<span class="op">=</span><span class="va">pred</span>,</span>
<span>                     ChIPScore<span class="op">=</span><span class="va">chip</span>,</span>
<span>                     chromatinState<span class="op">=</span><span class="va">Access</span>,</span>
<span>                     occupancy<span class="op">=</span><span class="va">occup</span>,</span>
<span>                     goodnessOfFit<span class="op">=</span><span class="va">accu</span>,</span>
<span>                     geneRef<span class="op">=</span><span class="va">geneRef</span><span class="op">)</span></span></code></pre></div>
<p><img src="ChIPanalyser_files/figure-html/unnamed-chunk-18-1.png" width="1152"><img src="ChIPanalyser_files/figure-html/unnamed-chunk-18-2.png" width="1152"><img src="ChIPanalyser_files/figure-html/unnamed-chunk-18-3.png" width="1152"><img src="ChIPanalyser_files/figure-html/unnamed-chunk-18-4.png" width="1152"></p>
<p>In this case we have also added a gene reference object. This object
is a GRanges object containing the postion of various <em>genetic
elements</em> such as 3’UTR, 5’UTR, introns , etc</p>
<p>NOTE <code>plotOccupancyProfile</code> offers the possibility to
customise graphical parameters. Unfortunately,
<code>plotOptimalHeatMaps</code> offers limited graphical parameter
customisation.</p>
</div>
</div>
<div class="section level3">
<h3 id="parameter-description">Parameter Description<a class="anchor" aria-label="anchor" href="#parameter-description"></a>
</h3>
<p>In the following section, we will describe the different parameters
present in both <code>parameterOptions</code> and
<code>genomicProfiles</code>. Information concerning arguments to
functions are described in the manual pages for each function.</p>
<p>As a reminder, here are the parameter options for the
<code>parameterOptions</code> object. Parameters are divided into
different categories depending on when they are required internally.</p>
<div class="sourceCode" id="cb57"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/parameterOptions.html">parameterOptions</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ </span></span>
<span><span class="co">##                         parameterOptions </span></span>
<span><span class="co">## _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ </span></span>
<span><span class="co">##  processingChIP options </span></span>
<span><span class="co">##  </span></span>
<span><span class="co">##  chipMean:                                                  200 </span></span>
<span><span class="co">##  chipSd:                                                    200 </span></span>
<span><span class="co">##  chipSmooth:                                                250 </span></span>
<span><span class="co">##  lociWidth:                                               20000 </span></span>
<span><span class="co">##  noiseFilter:                                              zero </span></span>
<span><span class="co">##  </span></span>
<span><span class="co">## _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ </span></span>
<span><span class="co">##  processingChIP options Updated </span></span>
<span><span class="co">##  </span></span>
<span><span class="co">##  maxSignal:                                                   1 </span></span>
<span><span class="co">##  backgroundSignal:                                            0 </span></span>
<span><span class="co">##  </span></span>
<span><span class="co">## _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ </span></span>
<span><span class="co">##  Position Weight Matrix Options </span></span>
<span><span class="co">##  </span></span>
<span><span class="co">##  naturalLog:                                               TRUE </span></span>
<span><span class="co">##  noOfSites:                                                 all </span></span>
<span><span class="co">##  PWMpseudocount:                                              1 </span></span>
<span><span class="co">##  </span></span>
<span><span class="co">## _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ </span></span>
<span><span class="co">##  Genome Wide Score options </span></span>
<span><span class="co">##  </span></span>
<span><span class="co">##  strandRule:                                                max </span></span>
<span><span class="co">##  whichstrand:                                                +- </span></span>
<span><span class="co">##  lambdaPWM:                                                   1 </span></span>
<span><span class="co">##  </span></span>
<span><span class="co">## _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ </span></span>
<span><span class="co">##  PWM Scores above Threshold options </span></span>
<span><span class="co">##  </span></span>
<span><span class="co">##  strandRule:                                                max </span></span>
<span><span class="co">##  whichstrand:                                                +- </span></span>
<span><span class="co">##  </span></span>
<span><span class="co">## _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ </span></span>
<span><span class="co">##  Occupancy options </span></span>
<span><span class="co">##  </span></span>
<span><span class="co">##  Ploidy:                                                      2 </span></span>
<span><span class="co">##  lambdaPWM:                                                   1 </span></span>
<span><span class="co">##  boundMolecules:                                           1000 </span></span>
<span><span class="co">##  maxSignal:                                                   1 </span></span>
<span><span class="co">##  backgroundSignal:                                            0 </span></span>
<span><span class="co">##  </span></span>
<span><span class="co">## _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ </span></span>
<span><span class="co">##  ChIP Profile options </span></span>
<span><span class="co">##  </span></span>
<span><span class="co">##  chipMean:                                                  200 </span></span>
<span><span class="co">##  chipSd:                                                    200 </span></span>
<span><span class="co">##  chipSmooth:                                                250 </span></span>
<span><span class="co">##  stepSize:                                                   10 </span></span>
<span><span class="co">##  </span></span>
<span><span class="co">## _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _</span></span></code></pre>
<ul>
<li>
<strong>chipMean</strong>: Average ChIP peak width. Peak width is
used during the smoothing of ChIP data.</li>
<li>
<strong>chipSd</strong>: Standard deviation of peak width. SD of
peak width is used during the smoothing of ChIP data.</li>
<li>
<strong>chipSmooth</strong>: Window width used for ChIP data
smoothing.</li>
<li>
<strong>lociWidth</strong>: When no loci are provided, ChIPanalyser
will split ChIP data into bins of width equals to lociWidth.</li>
<li>
<strong>noiseFilter</strong>: Noise filter applied to ChIP data.
ChIPanalyser provides four filters (zero, mean, median and sigmoid).
Zero assigns 0 to any score below zero. Mean and median assign 0 to any
score below the mean or median score. Sigmoid applies a weight to each
ChIP score based on a logistic distribution. Scores above the 95th
quantile will be weighted with a score between 1 and 2. Scores below the
95th quantile will be weighted with a score between 1 and 0.</li>
<li>
<strong>maxSignal</strong>: Maximum ChIP score after normalisation.
Required in later step of the analysis. However, this score is computed
internally and stored into the <code>ChIPscore</code> object , result of
<code>processingChIP</code>.</li>
<li>
<strong>backgroundSignal</strong>: Average ChIP score after
normalisation. Required in later step of the analysis. However, this
score is computed internally and stored into the <code>ChIPscore</code>
object , result of <code>processingChIP</code>.</li>
<li>
<strong>naturalLog</strong>: Log transform to be applied to PFM to
PWM conversion. If <code>TRUE</code>, natural log will be used otherwise
log2 will be used.</li>
<li>
<strong>noOfSites</strong>: Number of sites in the PWM that will be
used for analysis. Default is set at “all” meaning all sites will be
used. In the case that this argument is changed, ChIPanalyser requires a
numeric value describing the number of sites selected (from first
site).</li>
<li>
<strong>PWMpseudocount</strong>: Pseudo-count used during PFM to PWM
conversion.</li>
<li>
<strong>strandRule</strong>: PWM score computation mode.
<code>max</code> returns the highest PWM score regardless of strand.
<code>sum</code> returns the sum of PWM scores over both strands.
<code>mean</code> returns the average PWM score oer both strands. It
should be noted that this argument is only relevant when both strands
are considered. See below.</li>
<li>
<strong>whichstrand</strong>: Strand that should be used for
analysis. Options are: both <code>+-</code> or <code>-+</code>, plus
only <code>+</code> or negative only <code>-</code>.</li>
<li>
<strong>lambdaPWM</strong>: Value to be assigned to <span class="math inline">\(\lambda\)</span> . Positive numeric value.</li>
<li>
<strong>ploidy</strong>: Ploidy level of the organim used during
analysis.</li>
<li>
<strong>boundMolecules</strong>: Number of molecules used to run
analysis. Positive numeric value.</li>
</ul>
<p>As you can see, some of these parameters are used during multiple
steps of the analysis. If these paremeters have been changed in either a
<code>parameterOptions</code> object or <code>genomicProfiles</code>
object, please ensure that you parse these objects to each function.
Each function will extract the values you have assigned to each
parameter and use those values for analysis. It is possible to update,
these parameters between each step of the analysis however, we recommend
to set all parameters beforehand to avoid unwanted parameter
mismatch.</p>
<p>Next, we will describe the content of <code>genomicProfiles</code>
objects. As a reminder, <code>genomicProfiles</code> object have the
following structure:</p>
<div class="sourceCode" id="cb59"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/str.html" class="external-link">str</a></span><span class="op">(</span><span class="fu"><a href="../reference/genomicProfiles.html">genomicProfiles</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Formal class 'genomicProfiles' [package "ChIPanalyser"] with 31 slots</span></span>
<span><span class="co">##   ..@ PWM               : num[0 , 0 ] </span></span>
<span><span class="co">##   ..@ PFM               : num[0 , 0 ] </span></span>
<span><span class="co">##   ..@ PFMFormat         : chr "raw"</span></span>
<span><span class="co">##   ..@ BPFrequency       : num [1:4] 0.25 0.25 0.25 0.25</span></span>
<span><span class="co">##   ..@ minPWMScore       : logi(0) </span></span>
<span><span class="co">##   ..@ maxPWMScore       : logi(0) </span></span>
<span><span class="co">##   ..@ profiles          :Formal class 'CompressedGRangesList' [package "GenomicRanges"] with 5 slots</span></span>
<span><span class="co">##   .. .. ..@ unlistData     :Formal class 'GRanges' [package "GenomicRanges"] with 7 slots</span></span>
<span><span class="co">##   .. .. .. .. ..@ seqnames       :Formal class 'Rle' [package "S4Vectors"] with 4 slots</span></span>
<span><span class="co">##   .. .. .. .. .. .. ..@ values         : Factor w/ 0 levels: </span></span>
<span><span class="co">##   .. .. .. .. .. .. ..@ lengths        : int(0) </span></span>
<span><span class="co">##   .. .. .. .. .. .. ..@ elementMetadata: NULL</span></span>
<span><span class="co">##   .. .. .. .. .. .. ..@ metadata       : list()</span></span>
<span><span class="co">##   .. .. .. .. ..@ ranges         :Formal class 'IRanges' [package "IRanges"] with 6 slots</span></span>
<span><span class="co">##   .. .. .. .. .. .. ..@ start          : int(0) </span></span>
<span><span class="co">##   .. .. .. .. .. .. ..@ width          : int(0) </span></span>
<span><span class="co">##   .. .. .. .. .. .. ..@ NAMES          : NULL</span></span>
<span><span class="co">##   .. .. .. .. .. .. ..@ elementType    : chr "ANY"</span></span>
<span><span class="co">##   .. .. .. .. .. .. ..@ elementMetadata: NULL</span></span>
<span><span class="co">##   .. .. .. .. .. .. ..@ metadata       : list()</span></span>
<span><span class="co">##   .. .. .. .. ..@ strand         :Formal class 'Rle' [package "S4Vectors"] with 4 slots</span></span>
<span><span class="co">##   .. .. .. .. .. .. ..@ values         : Factor w/ 3 levels "+","-","*": </span></span>
<span><span class="co">##   .. .. .. .. .. .. ..@ lengths        : int(0) </span></span>
<span><span class="co">##   .. .. .. .. .. .. ..@ elementMetadata: NULL</span></span>
<span><span class="co">##   .. .. .. .. .. .. ..@ metadata       : list()</span></span>
<span><span class="co">##   .. .. .. .. ..@ seqinfo        :Formal class 'Seqinfo' [package "GenomeInfoDb"] with 4 slots</span></span>
<span><span class="co">##   .. .. .. .. .. .. ..@ seqnames   : chr(0) </span></span>
<span><span class="co">##   .. .. .. .. .. .. ..@ seqlengths : int(0) </span></span>
<span><span class="co">##   .. .. .. .. .. .. ..@ is_circular: logi(0) </span></span>
<span><span class="co">##   .. .. .. .. .. .. ..@ genome     : chr(0) </span></span>
<span><span class="co">##   .. .. .. .. ..@ elementMetadata:Formal class 'DFrame' [package "S4Vectors"] with 6 slots</span></span>
<span><span class="co">##   .. .. .. .. .. .. ..@ rownames       : NULL</span></span>
<span><span class="co">##   .. .. .. .. .. .. ..@ nrows          : int 0</span></span>
<span><span class="co">##   .. .. .. .. .. .. ..@ elementType    : chr "ANY"</span></span>
<span><span class="co">##   .. .. .. .. .. .. ..@ elementMetadata: NULL</span></span>
<span><span class="co">##   .. .. .. .. .. .. ..@ metadata       : list()</span></span>
<span><span class="co">##   .. .. .. .. .. .. ..@ listData       : Named list()</span></span>
<span><span class="co">##   .. .. .. .. ..@ elementType    : chr "ANY"</span></span>
<span><span class="co">##   .. .. .. .. ..@ metadata       : list()</span></span>
<span><span class="co">##   .. .. ..@ elementMetadata:Formal class 'DFrame' [package "S4Vectors"] with 6 slots</span></span>
<span><span class="co">##   .. .. .. .. ..@ rownames       : NULL</span></span>
<span><span class="co">##   .. .. .. .. ..@ nrows          : int 0</span></span>
<span><span class="co">##   .. .. .. .. ..@ elementType    : chr "ANY"</span></span>
<span><span class="co">##   .. .. .. .. ..@ elementMetadata: NULL</span></span>
<span><span class="co">##   .. .. .. .. ..@ metadata       : list()</span></span>
<span><span class="co">##   .. .. .. .. ..@ listData       : Named list()</span></span>
<span><span class="co">##   .. .. ..@ elementType    : chr "GRanges"</span></span>
<span><span class="co">##   .. .. ..@ metadata       : list()</span></span>
<span><span class="co">##   .. .. ..@ partitioning   :Formal class 'PartitioningByEnd' [package "IRanges"] with 5 slots</span></span>
<span><span class="co">##   .. .. .. .. ..@ end            : int(0) </span></span>
<span><span class="co">##   .. .. .. .. ..@ NAMES          : NULL</span></span>
<span><span class="co">##   .. .. .. .. ..@ elementType    : chr "ANY"</span></span>
<span><span class="co">##   .. .. .. .. ..@ elementMetadata: NULL</span></span>
<span><span class="co">##   .. .. .. .. ..@ metadata       : list()</span></span>
<span><span class="co">##   ..@ DNASequenceLength : logi(0) </span></span>
<span><span class="co">##   ..@ averageExpPWMScore: logi(0) </span></span>
<span><span class="co">##   ..@ ZeroBackground    : logi(0) </span></span>
<span><span class="co">##   ..@ drop              : logi(0) </span></span>
<span><span class="co">##   ..@ tags              : chr "empty"</span></span>
<span><span class="co">##   ..@ ploidy            : num 2</span></span>
<span><span class="co">##   ..@ boundMolecules    : num 1000</span></span>
<span><span class="co">##   ..@ backgroundSignal  : num 0</span></span>
<span><span class="co">##   ..@ maxSignal         : num 1</span></span>
<span><span class="co">##   ..@ lociWidth         : num 20000</span></span>
<span><span class="co">##   ..@ chipMean          : num 200</span></span>
<span><span class="co">##   ..@ chipSd            : num 200</span></span>
<span><span class="co">##   ..@ chipSmooth        : num 250</span></span>
<span><span class="co">##   ..@ stepSize          : num 10</span></span>
<span><span class="co">##   ..@ removeBackground  : num 0</span></span>
<span><span class="co">##   ..@ noiseFilter       : chr "zero"</span></span>
<span><span class="co">##   ..@ PWMThreshold      : num 0.7</span></span>
<span><span class="co">##   ..@ strandRule        : chr "max"</span></span>
<span><span class="co">##   ..@ whichstrand       : chr "+-"</span></span>
<span><span class="co">##   ..@ lambdaPWM         : num 1</span></span>
<span><span class="co">##   ..@ naturalLog        : logi TRUE</span></span>
<span><span class="co">##   ..@ noOfSites         : chr "all"</span></span>
<span><span class="co">##   ..@ PWMpseudocount    : num 1</span></span>
<span><span class="co">##   ..@ paramTag          : chr "empty"</span></span></code></pre>
<p>As you can see, we are using the structure function to show all
internal slots. The <code>genomicProfiles</code> object inherit from
<code>parameterOptions</code>, contain slots that are not user
updatetable and finally the show method applied to
<code>genomicProfiles</code> varies with each step of the analysis. This
is intended to reduce information overload when “looking” at an
object.</p>
<ul>
<li>
<strong>PWM</strong>: A Position Weight Matrix. Either directly user
provided or will be built internally if a Position Frequency Matrix is
provided.</li>
<li>
<strong>PFM</strong>: A Position Frequency Matrix. This argument can
either be a path towards a file containing the PFM (RAW, JASPAR or
Transfac format) or a matrix with rows being A, C, T, G and columns
being PFM sites.</li>
<li>
<strong>PFMFormat</strong>: Format of provided PFM.
<code>PFMFormat</code> can be one of the following: RAW, JASPAR,
transfac or matrix. Matrix format is used if the provided PFM is an R
matrix.</li>
<li>
<strong>BPFrequency</strong>: Genome alphabet Frequency. This
parameter can be user provided in the form of a numeric vector
(Frequency of each base pair in the following order A, C, T, G). For
convenience, ChIPanalyser will automatically compute genome wide base
pair frequency if a <code>DNAStringSet</code> object or
<code>BSgenome</code> object is provided to this argument.</li>
<li>
<strong>minPWMScore</strong>: minimum PWM score over entire genome.
Internally updated.</li>
<li>
<strong>maxPWMScore</strong>: maximum PWM score over entire genome.
Internally updated.</li>
<li>
<strong>profiles</strong>: Storage slot for PWM scores above
threshold, Occupancy, ChIP like profiles and Goodness of Fit. This slot
holds the results each step of the analysis. Updated internally.</li>
<li>
<strong>DNASequenceLength</strong>: Length of DNA sequence used for
analysis. Updated internally but can be provided by user.</li>
<li>
<strong>averageExpPWMScore</strong>: Average Exponetial PWM score.
Updated Internally.</li>
<li>
<strong>drop</strong>: Regions that did not contain any accessible
DNA or did not contain sites above threshold.</li>
</ul>
<p>All other parameters have either been explained above or are part of
the internal working of the package. These parameters are mainly used to
keep track of the advancement of the analysis between each step. They
should not be changed by user.</p>
<p>Finally, we provide a list of setter/getter functions for each
slot:</p>
<div class="sourceCode" id="cb61"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span>    <span class="co">## Accessors and Setters for parameterOptions and genomicProfiles</span></span>
<span>    <span class="fu">avrageExpPWMScore</span><span class="op">(</span><span class="va">obj</span><span class="op">)</span></span>
<span>    <span class="fu"><a href="../reference/backgroundSignal.html">backgroundSignal</a></span><span class="op">(</span><span class="va">obj</span><span class="op">)</span></span>
<span>    <span class="fu"><a href="../reference/backgroundSignal.html">backgroundSignal</a></span><span class="op">(</span><span class="va">obj</span><span class="op">)</span><span class="op">&lt;-</span><span class="va">value</span></span>
<span>    <span class="fu"><a href="../reference/boundMolecules.html">boundMolecules</a></span><span class="op">(</span><span class="va">obj</span><span class="op">)</span></span>
<span>    <span class="fu"><a href="../reference/boundMolecules.html">boundMolecules</a></span><span class="op">(</span><span class="va">obj</span><span class="op">)</span><span class="op">&lt;-</span><span class="va">value</span></span>
<span>    <span class="fu"><a href="../reference/BPFrequency.html">BPFrequency</a></span><span class="op">(</span><span class="va">obj</span><span class="op">)</span></span>
<span>    <span class="fu"><a href="../reference/BPFrequency.html">BPFrequency</a></span><span class="op">(</span><span class="va">obj</span><span class="op">)</span><span class="op">&lt;-</span><span class="va">value</span></span>
<span>    <span class="fu"><a href="../reference/chipMean.html">chipMean</a></span><span class="op">(</span><span class="va">obj</span><span class="op">)</span></span>
<span>    <span class="fu"><a href="../reference/chipMean.html">chipMean</a></span><span class="op">(</span><span class="va">obj</span><span class="op">)</span><span class="op">&lt;-</span><span class="va">value</span></span>
<span>    <span class="fu"><a href="../reference/chipSd.html">chipSd</a></span><span class="op">(</span><span class="va">obj</span><span class="op">)</span></span>
<span>    <span class="fu"><a href="../reference/chipSd.html">chipSd</a></span><span class="op">(</span><span class="va">obj</span><span class="op">)</span><span class="op">&lt;-</span><span class="va">value</span></span>
<span>    <span class="fu"><a href="../reference/chipSmooth.html">chipSmooth</a></span><span class="op">(</span><span class="va">obj</span><span class="op">)</span></span>
<span>    <span class="fu"><a href="../reference/chipSmooth.html">chipSmooth</a></span><span class="op">(</span><span class="va">obj</span><span class="op">)</span><span class="op">&lt;-</span><span class="va">value</span></span>
<span>    <span class="fu"><a href="../reference/DNASequenceLength.html">DNASequenceLength</a></span><span class="op">(</span><span class="va">obj</span><span class="op">)</span></span>
<span>    <span class="fu"><a href="../reference/drop.html">drop</a></span><span class="op">(</span><span class="va">obj</span><span class="op">)</span></span>
<span>    <span class="fu"><a href="../reference/lambdaPWM.html">lambdaPWM</a></span><span class="op">(</span><span class="va">obj</span><span class="op">)</span></span>
<span>    <span class="fu"><a href="../reference/lambdaPWM.html">lambdaPWM</a></span><span class="op">(</span><span class="va">obj</span><span class="op">)</span><span class="op">&lt;-</span><span class="va">value</span></span>
<span>    <span class="fu"><a href="../reference/lociWidth.html">lociWidth</a></span><span class="op">(</span><span class="va">obj</span><span class="op">)</span></span>
<span>    <span class="fu"><a href="../reference/lociWidth.html">lociWidth</a></span><span class="op">(</span><span class="va">obj</span><span class="op">)</span><span class="op">&lt;-</span><span class="va">value</span></span>
<span>    <span class="fu"><a href="../reference/maxPWMScore.html">maxPWMScore</a></span><span class="op">(</span><span class="va">obj</span><span class="op">)</span></span>
<span>    <span class="fu"><a href="../reference/maxSignal.html">maxSignal</a></span><span class="op">(</span><span class="va">obj</span><span class="op">)</span></span>
<span>    <span class="fu"><a href="../reference/maxSignal.html">maxSignal</a></span><span class="op">(</span><span class="va">obj</span><span class="op">)</span><span class="op">&lt;-</span><span class="va">value</span></span>
<span>    <span class="fu"><a href="../reference/minPWMScore.html">minPWMScore</a></span><span class="op">(</span><span class="va">obj</span><span class="op">)</span></span>
<span>    <span class="fu"><a href="../reference/naturalLog.html">naturalLog</a></span><span class="op">(</span><span class="va">obj</span><span class="op">)</span></span>
<span>    <span class="fu"><a href="../reference/naturalLog.html">naturalLog</a></span><span class="op">(</span><span class="va">obj</span><span class="op">)</span><span class="op">&lt;-</span><span class="va">value</span></span>
<span>    <span class="fu"><a href="../reference/noiseFilter.html">noiseFilter</a></span><span class="op">(</span><span class="va">obj</span><span class="op">)</span></span>
<span>    <span class="fu"><a href="../reference/noiseFilter.html">noiseFilter</a></span><span class="op">(</span><span class="va">obj</span><span class="op">)</span><span class="op">&lt;-</span><span class="va">value</span></span>
<span>    <span class="fu"><a href="../reference/noOfSites.html">noOfSites</a></span><span class="op">(</span><span class="va">obj</span><span class="op">)</span></span>
<span>    <span class="fu"><a href="../reference/noOfSites.html">noOfSites</a></span><span class="op">(</span><span class="va">obj</span><span class="op">)</span><span class="op">&lt;-</span><span class="va">value</span></span>
<span>    <span class="fu"><a href="../reference/PFMFormat.html">PFMFormat</a></span><span class="op">(</span><span class="va">obj</span><span class="op">)</span></span>
<span>    <span class="fu"><a href="../reference/PFMFormat.html">PFMFormat</a></span><span class="op">(</span><span class="va">obj</span><span class="op">)</span><span class="op">&lt;-</span><span class="va">value</span></span>
<span>    <span class="fu"><a href="../reference/ploidy.html">ploidy</a></span><span class="op">(</span><span class="va">obj</span><span class="op">)</span></span>
<span>    <span class="fu"><a href="../reference/ploidy.html">ploidy</a></span><span class="op">(</span><span class="va">obj</span><span class="op">)</span><span class="op">&lt;-</span><span class="va">value</span></span>
<span>    <span class="fu"><a href="../reference/PositionFrequencyMatrix.html">PositionFrequencyMatrix</a></span><span class="op">(</span><span class="va">obj</span><span class="op">)</span></span>
<span>    <span class="fu"><a href="../reference/PositionFrequencyMatrix.html">PositionFrequencyMatrix</a></span><span class="op">(</span><span class="va">obj</span><span class="op">)</span><span class="op">&lt;-</span><span class="va">value</span></span>
<span>    <span class="fu"><a href="../reference/PositionWeightMatrix.html">PositionWeightMatrix</a></span><span class="op">(</span><span class="va">obj</span><span class="op">)</span></span>
<span>    <span class="fu"><a href="../reference/PositionWeightMatrix.html">PositionWeightMatrix</a></span><span class="op">(</span><span class="va">obj</span><span class="op">)</span><span class="op">&lt;-</span><span class="va">value</span></span>
<span>    <span class="fu"><a href="../reference/profiles-methods.html">profiles</a></span><span class="op">(</span><span class="va">obj</span><span class="op">)</span></span>
<span>    <span class="fu"><a href="../reference/PWMpseudocount.html">PWMpseudocount</a></span><span class="op">(</span><span class="va">obj</span><span class="op">)</span></span>
<span>    <span class="fu"><a href="../reference/PWMpseudocount.html">PWMpseudocount</a></span><span class="op">(</span><span class="va">obj</span><span class="op">)</span><span class="op">&lt;-</span><span class="va">value</span></span>
<span>    <span class="fu"><a href="../reference/PWMThreshold.html">PWMThreshold</a></span><span class="op">(</span><span class="va">obj</span><span class="op">)</span></span>
<span>    <span class="fu"><a href="../reference/PWMThreshold.html">PWMThreshold</a></span><span class="op">(</span><span class="va">obj</span><span class="op">)</span><span class="op">&lt;-</span><span class="va">value</span></span>
<span>    <span class="fu"><a href="../reference/removeBackground.html">removeBackground</a></span><span class="op">(</span><span class="va">obj</span><span class="op">)</span></span>
<span>    <span class="fu"><a href="../reference/removeBackground.html">removeBackground</a></span><span class="op">(</span><span class="va">obj</span><span class="op">)</span><span class="op">&lt;-</span><span class="va">value</span></span>
<span>    <span class="fu"><a href="../reference/stepSize.html">stepSize</a></span><span class="op">(</span><span class="va">obj</span><span class="op">)</span></span>
<span>    <span class="fu"><a href="../reference/stepSize.html">stepSize</a></span><span class="op">(</span><span class="va">obj</span><span class="op">)</span><span class="op">&lt;-</span><span class="va">value</span></span>
<span>    <span class="fu"><a href="../reference/strandRule.html">strandRule</a></span><span class="op">(</span><span class="va">obj</span><span class="op">)</span></span>
<span>    <span class="fu"><a href="../reference/strandRule.html">strandRule</a></span><span class="op">(</span><span class="va">obj</span><span class="op">)</span><span class="op">&lt;-</span><span class="va">value</span></span>
<span>    <span class="fu"><a href="../reference/whichstrand.html">whichstrand</a></span><span class="op">(</span><span class="va">obj</span><span class="op">)</span></span>
<span>    <span class="fu"><a href="../reference/whichstrand.html">whichstrand</a></span><span class="op">(</span><span class="va">obj</span><span class="op">)</span><span class="op">&lt;-</span><span class="va">value</span></span>
<span>    <span class="co">## ChIPScore slots accessors </span></span>
<span>    <span class="fu"><a href="../reference/loci.html">loci</a></span><span class="op">(</span><span class="va">obj</span><span class="op">)</span></span>
<span>    <span class="fu"><a href="../reference/scores.html">scores</a></span><span class="op">(</span><span class="va">obj</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="session-info">Session Info<a class="anchor" aria-label="anchor" href="#session-info"></a>
</h3>
<div class="sourceCode" id="cb62"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## R version 4.2.0 (2022-04-22)</span></span>
<span><span class="co">## Platform: x86_64-apple-darwin17.0 (64-bit)</span></span>
<span><span class="co">## Running under: macOS Big Sur/Monterey 10.16</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Matrix products: default</span></span>
<span><span class="co">## BLAS:   /Library/Frameworks/R.framework/Versions/4.2/Resources/lib/libRblas.0.dylib</span></span>
<span><span class="co">## LAPACK: /Library/Frameworks/R.framework/Versions/4.2/Resources/lib/libRlapack.dylib</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## locale:</span></span>
<span><span class="co">## [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## attached base packages:</span></span>
<span><span class="co">## [1] parallel  stats4    stats     graphics  grDevices utils     datasets </span></span>
<span><span class="co">## [8] methods   base     </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## other attached packages:</span></span>
<span><span class="co">##  [1] BSgenome.Dmelanogaster.UCSC.dm6_1.4.1 ChIPanalyser_2.0.1                   </span></span>
<span><span class="co">##  [3] RcppRoll_0.3.0                        BSgenome_1.64.0                      </span></span>
<span><span class="co">##  [5] rtracklayer_1.56.1                    Biostrings_2.64.0                    </span></span>
<span><span class="co">##  [7] XVector_0.36.0                        GenomicRanges_1.48.0                 </span></span>
<span><span class="co">##  [9] GenomeInfoDb_1.32.2                   IRanges_2.30.0                       </span></span>
<span><span class="co">## [11] S4Vectors_0.34.0                      BiocGenerics_0.42.0                  </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## loaded via a namespace (and not attached):</span></span>
<span><span class="co">##  [1] SummarizedExperiment_1.26.1 xfun_0.31                  </span></span>
<span><span class="co">##  [3] bslib_0.4.0                 purrr_0.3.4                </span></span>
<span><span class="co">##  [5] lattice_0.20-45             htmltools_0.5.3            </span></span>
<span><span class="co">##  [7] yaml_2.3.5                  XML_3.99-0.10              </span></span>
<span><span class="co">##  [9] rlang_1.0.4                 pkgdown_2.0.6              </span></span>
<span><span class="co">## [11] jquerylib_0.1.4             BiocParallel_1.30.3        </span></span>
<span><span class="co">## [13] RColorBrewer_1.1-3          matrixStats_0.62.0         </span></span>
<span><span class="co">## [15] GenomeInfoDbData_1.2.8      stringr_1.4.0              </span></span>
<span><span class="co">## [17] zlibbioc_1.42.0             MatrixGenerics_1.8.1       </span></span>
<span><span class="co">## [19] ragg_1.2.2                  codetools_0.2-18           </span></span>
<span><span class="co">## [21] memoise_2.0.1               evaluate_0.15              </span></span>
<span><span class="co">## [23] restfulr_0.0.15             Biobase_2.56.0             </span></span>
<span><span class="co">## [25] knitr_1.39                  fastmap_1.1.0              </span></span>
<span><span class="co">## [27] highr_0.9                   Rcpp_1.0.9                 </span></span>
<span><span class="co">## [29] ROCR_1.0-11                 BiocManager_1.30.18        </span></span>
<span><span class="co">## [31] cachem_1.0.6                DelayedArray_0.22.0        </span></span>
<span><span class="co">## [33] desc_1.4.1                  jsonlite_1.8.0             </span></span>
<span><span class="co">## [35] systemfonts_1.0.4           fs_1.5.2                   </span></span>
<span><span class="co">## [37] Rsamtools_2.12.0            textshaping_0.3.6          </span></span>
<span><span class="co">## [39] rjson_0.2.21                digest_0.6.29              </span></span>
<span><span class="co">## [41] stringi_1.7.8               BiocIO_1.6.0               </span></span>
<span><span class="co">## [43] rprojroot_2.0.3             grid_4.2.0                 </span></span>
<span><span class="co">## [45] cli_3.3.0                   tools_4.2.0                </span></span>
<span><span class="co">## [47] bitops_1.0-7                magrittr_2.0.3             </span></span>
<span><span class="co">## [49] sass_0.4.2                  RCurl_1.98-1.8             </span></span>
<span><span class="co">## [51] crayon_1.5.1                Matrix_1.4-1               </span></span>
<span><span class="co">## [53] rmarkdown_2.14              R6_2.5.1                   </span></span>
<span><span class="co">## [55] GenomicAlignments_1.32.1    compiler_4.2.0</span></span></code></pre>
</div>
<div class="section level3">
<h3 id="references">References<a class="anchor" aria-label="anchor" href="#references"></a>
</h3>
<p>Zabet NR, Adryan B (2015) Estimating binding properties of
transcription factors from genome-wide binding profiles. Nucleic Acids
Res., 43, 84–94.</p>
</div>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by Patrick CN Martin, Nicolea Radu Zabet.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.6.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
